add v5.3 release notes (#7369)
pingcap/docs-cn: releases/release-5.3.0.md

### 系统变量

|  变量名    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- |
|  |  |  |
|  |  |  |
|  |  |  |

### 配置文件参数

|  配置文件    |  配置项    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- | :----------- |
|  |  |  |
|  |  |  |
|  |  |  |

### 其他

- 
- 
- 
- 

## 新功能

### SQL

- **功能 1**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)

- **功能 2**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)    

### 事务

- **功能 3**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)    

### 字符集和排序规则

- **功能 4**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)   

### 安全

- **功能 5**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)   

## 性能优化

- **功能 6**

> 

Reviewer: @qiancai 我们最好能直接commit，不要在这里comment了... ##性能优化 和下面 ##稳定性提升 都包含在后面 ##提升改进 里的 TiFlash 部分里了。这里的或后面的可以只保留一处。

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 稳定性提升

- **功能 7**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

- **功能 8**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 高可用和容灾

- **功能 9**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 数据迁移

- **功能 10**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

>     <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

Reviewer 2: **[DM] 复制性能提高（Replication performance improvement）**

以更低的延迟将数据从 MySQL 同步到 TiDB，包含三项子内容

- 合并单行数据的多次变更（Compact multiple updates on a single row into one statement）
- 点查更新合并为批量操作（Merge batch updates of multiple rows into one statement）
- 异步保存检查点（Async Checkpoint）

**[DM] 增加 OpenAPI 以更方便的管理集群（Add OpenAPI to facilitate users to manage DM tasks）**

**[Lightning] 支持导入 GBK 编码文件 [文档](https://docs.pingcap.com/zh/tidb/dev/tidb-lightning-configuration)**

**[Lightning] 支持忽略部分错误行**

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## TiDB 数据共享订阅

- **功能 10**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 问题诊断效率

- **功能 10**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 部署及运维

> - **功能 10**

Reviewer 3: 持续性能分析
TiDB Dashboard 引入“持续性能分析”功能，提供在集群运行状态时自动保存实例性能分析结果的能力。
该功能默认关闭，需进入 TiDB Dashboard 持续性能分析页面开启。
“持续性能分析”功能必须使用 TiUP 1.7.0 及以上版本升级/安装的集群才可使用。

Author: 内容已更新 PTAL 谢谢

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)    

### 事务

- **功能 3**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)    

### 字符集和排序规则

- **功能 4**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)   

### 安全

- **功能 5**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)   

## 性能优化

- **功能 6**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 稳定性提升

- **功能 7**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

- **功能 8**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 高可用和容灾

- **功能 9**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

> ## 数据迁移

Reviewer 4: 新版本 TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。 该功能无须特别的配置，在不改变用户使用习惯的同时，极大的提高用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。
    在我们性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。

## 高可用和容灾

- **功能 9**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 数据迁移

- **支持部署多个 Lightning**

    新版本 TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。 该功能无须特别的配置，在不改变用户使用习惯的同时，极大的提高用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。

    在我们性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。（数据待更新）

    [用户文档](/)  

- **提高 DM 复制性能（Replication performance improvement）**

    以更低的延迟将数据从 MySQL 同步到 TiDB，包含三项子内容

    - 合并单行数据的多次变更（Compact multiple updates on a single row into one statement）
    - 点查更新合并为批量操作（Merge batch updates of multiple rows into one statement）
    - 异步保存检查点（Async Checkpoint）

- **增加 DM 的 OpenAPI 以更方便地管理集群**

- **Lightning 支持导入 GBK 编码文件** 

    [用户文档](/tidb-lightning-configuration.md)

- **Lightning 支持忽略部分错误行**

## TiDB 数据共享订阅

- **功能 10**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 问题诊断效率

- **功能 10**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 部署及运维

> - **持续性能分析**

Reviewer 3: ```suggestion
- **持续性能分析（实验特性）**
```

## 性能优化

- **提升 TiFlash 实时分析能力**

    - 降低远端数据读取的开销，减少网络传输量
    - 大幅优化 TiFlash TableScan 算子的执行效率
    - 新增更多的函数支持，提升
        - 新增运算符支持：LIKE expression
        - 新增若干字符串函数支持：FORMAT(), LOWER(), LTRIM(), RTRIM(), SUBSTRING_INDEX(), TRIM(), UCASE(), UPPER()
        - 新增数学函数支持：ROUND(decimal, int)
        - 新增若干日期时间函数支持：HOUR(), MICROSECOND(), MINUTE(), SECOND(), SYSDATE()
        - 新增 CAST 函数支持：CAST(time, real)
        - 新增若干聚合函数支持：GROUP_CONCAT(), SUM(enum)

    - 支持在非 Linux 平台上，使用 Dashboard 查看硬件信息

    [用户文档](/)  

## 稳定性提升

- **提升 TiFlash 稳定性**

    - 优化在高负载下查询容易超时的问题
    - 优化 TiFlash 日志搜索性能，避免搜索大体量日志（大于 10GB）时出现的卡顿或失败现象
    - 加强数据历史版本的回收策略
    - 使用 TiUP 重启或升级多个 TiFlash 节点时，提升了滚动重启 TiFlash 进程过程中的服务稳定性

- **功能 8**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 高可用和容灾

- **功能 9**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 数据迁移

- **支持部署多个 TiDB Lightning**

    新版本 TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。 该功能无需特别的配置，在不改变用户使用习惯的同时，极大提高了用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。

>     在产品性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。（数据待更新）

Reviewer 4: ```suggestion
    在产品性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。（数据待更新）。此外在 MySQL 分表数据聚合迁移到 TiDB 的场景中，MySQL 分表之间可能有冲突数据（主键/唯一键索引相同的数据），Lightning 也支持了冲突数据监测功能，用户可以使用该功能找到冲突数据，然后按照业务需要进行处理，冲突检测使用文档。
```

>     在产品性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。（数据待更新）。此外在 MySQL 分表数据聚合迁移到 TiDB 的场景中，MySQL 分表之间可能有冲突数据（主键/唯一键索引相同的数据），Lightning 也支持了冲突数据监测功能，用户可以使用该功能找到冲突数据，然后按照业务需要进行处理，冲突检测使用文档。

Reviewer 4: ```suggestion
    在产品性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。（数据待更新）。此外在 MySQL 分表数据聚合迁移到 TiDB 的场景中，MySQL 分表之间可能有冲突数据（主键/唯一键索引相同的数据），Lightning 也支持了数据导入过程中检查冲突数据的功能，用户可以使用该功能发现冲突数据，然后按照业务规则进行处理，冲突检测使用文档（待更新）。
```

---
title: TiDB 5.3 Release Notes
---

# TiDB 5.3 Release Notes

发版日期：2021 年 11 月 16 日

TiDB 版本：5.3.0

在 5.3 版本中，你可以获得以下关键特性：

+ 
+ 
+ 

## 兼容性变化

> **注意：**
>
> 当从一个早期的 TiDB 版本升级到 TiDB 5.3.0 时，如需了解所有中间版本对应的兼容性更改说明，请查看对应版本的 [Release Note](/releases/release-notes.md)。

### 系统变量

|  变量名    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- |

> |tidb_enable_noop_functions  | 修改 | 此变量不在控制 `CREATE TEMPORARY TABLE` 和 `DROP TEMPORARY TABLE` 行为。 |

Reviewer 5: ```suggestion
|tidb_enable_noop_functions  | 修改 | 此变量不再控制 `CREATE TEMPORARY TABLE` 和 `DROP TEMPORARY TABLE` 行为。 |
```

| tidb_tmp_table_max_size | 新增  | 此变量用于限制单个[临时表](/temporary-table.md)的最大大小，临时表超出该大小后报错。 |
|  |  |  |

### 配置文件参数

|  配置文件    |  配置项    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- | :----------- |
|  |  |  |
|  |  |  |
|  |  |  |

### 其他

- 临时表：

>     - 如果在 v5.3.0 升级前创建了 session 临时表，这些临时表在升级后会被 TiDB 当成普通表处理，导致应用报错。在 v5.3.0 上创建的 global 临时表在降级后会被当作普通表处理，导致应用报错。

Reviewer 6: ```suggestion
    - 如果在 v5.3.0 升级前创建了本地临时表，这些临时表实际为普通表，在升级后也会被 TiDB 当成普通表处理。在 v5.3.0 上创建的全局临时表在降级后会被当作普通表处理，导致数据错误。
```

>     - TiCDC 和 BR 从 v5.3.0 开始支持[全局临时表]((/temporary-table.md#全局临时表)。如果使用 v5.3.0 以下版本同步全局临时表到下游，会导致表定义错误。 如果 TiCDC 和 BR 的上游集群包含全局临时表，下游集群必须是 TiDB 5.3.0 及以上版本，否则同步报错。

Reviewer 6: ```suggestion
    - TiCDC 和 BR 从 v5.3.0 开始支持[全局临时表](/temporary-table.md#全局临时表)。如果使用 v5.3.0 以下版本同步全局临时表到下游，会导致表定义错误。 
    - 如果 TiCDC、TiDB-Binlog 的上游集群、 BR 的备份集群、Lightning 的源文件包含全局临时表，下游集群必须是 TiDB v5.3.0 及以上版本，否则同步报错。
```

>     - 更多关于临时表兼容性信息，请参考 [临时表与其他功能的兼容性](/temporary-table.md#与其他功能的兼容性)

Reviewer 6: ```suggestion
    - 更多关于临时表兼容性信息，请参考 [临时表与其他功能的兼容性](/temporary-table.md#与其他-tidb-功能的兼容性限制)
```

    - 点查更新合并为批量操作（Merge batch updates of multiple rows into one statement）
    - 异步保存检查点（Async Checkpoint）

- **增加 DM 的 OpenAPI 以更方便地管理集群**

    <功能描述 （DM 的 OpenAPI 是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

- **Lightning 支持导入 GBK 编码文件** 

    [用户文档](/tidb-lightning-configuration.md)

- **Lightning 支持忽略部分错误行**

    <能给用户带来什么好处>

## TiDB 数据共享订阅

- **功能 10**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 问题诊断效率

- **功能 10**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

## 部署及运维

- **持续性能分析**

    TiDB Dashboard 引入持续性能分析功能，提供在集群运行状态时自动保存实例性能分析结果的能力，通过火焰图的形式提高了 TiDB 集群性能的可观测性，有助于缩短故障诊断时间。

    该功能默认关闭，需进入 TiDB Dashboard 持续性能分析页面开启。

    持续性能分析功能必须使用 TiUP 1.7.0 及以上版本升级或安装的集群才可使用。

    [用户文档](/)  

## 遥测

TiDB 在遥测中新增收集 <列出本次新增遥测内容>。

若要了解所收集的信息详情及如何禁用该行为，请参见[遥测](/telemetry.md)文档。

## 提升改进

+ TiDB

    - 当 coprocessor 遇到锁时，在调试日志中显示受影响的 SQL 语句帮助诊断问题 [#27718](https://github.com/pingcap/tidb/issues/27718)

>     - 在 SQL 逻辑层备份和恢复数据时，支持显示备份和恢复数据的大小 [#27247](https://github.com/pingcap/tidb/issues/27247)

Reviewer 7: ```suggestion
    - 在 SQL 逻辑层备份和恢复数据时，支持显示备份和恢复数据的大小 [#27247](https://github.com/pingcap/tidb/issues/27247)
    - 改进 `tidb_analyze_version=2` 时 ANALYZE 默认的收集逻辑，提高收集速度并且降低资源开销
    - 引入语法 `ANALYZE TABLE table_name COLUMNS col_1, col_2, ..., col_n`，为宽表提供只收集一部分列统计信息的方案。提高宽表收集统计信息的速度。
```

+ TiKV

    - 简化 L0 层流控算法 [#10879](https://github.com/tikv/tikv/issues/10879)

>     - 优化 raft client 错误日志的收集 [#10983](https://github.com/tikv/tikv/pull/10983)

Reviewer 8: ```suggestion
    - 优化 raft client 错误日志的收集 [#10944](https://github.com/tikv/tikv/pull/10944)
```

    - 移除对基于 3DES (三重数据加密算法) 不安全的 TLS 加密套件的支持 [#27859](https://github.com/pingcap/tidb/pull/27859)
    - 修复因 Lightning 前置检查会连接已下线的 TiKV 节点而导致导入失败的问题 [#27826](https://github.com/pingcap/tidb/pull/27826)
    - 修复在导入太多文件到表时前置检查花费太多时间的问题 [#27605](https://github.com/pingcap/tidb/issues/27605)
    - 修复表达式重写时 `between` 推断出错误排序规则的问题 [#27146](https://github.com/pingcap/tidb/issues/27146)
    - 修复 `group_concat` 函数没有考虑排序规则的问题 [#27429](https://github.com/pingcap/tidb/issues/27429)
    - 修复 `extract` 函数处理负值时的问题 [#27236](https://github.com/pingcap/tidb/issues/27236)
    - 修复当设置 `NO_UNSIGNED_SUBTRACTION` 时创建分区失败的问题 [#26765](https://github.com/pingcap/tidb/issues/26765)
    - 避免在列修剪和聚合下推中使用有副作用的表达式 [#27106](https://github.com/pingcap/tidb/issues/27106)
    - 删除无用的 gRPC 日志 [#24190](https://github.com/pingcap/tidb/issues/24190)
    - 限制有效的小数点长度以修复精度相关的问题 [3091](https://github.com/pingcap/tics/issues/3091)
    - 修复 `plus` 表达式中检查溢出方法出错的问题 [26977](https://github.com/pingcap/tidb/issues/26977)
    - 修复当导出带有 `new collation` 数据的表的统计信息时报 `data too long` 错误的问题 [27024](https://github.com/pingcap/tidb/issues/27024)
    - 修复 `TIDB_TRX` 中不包含重试事务的问题 [28670](https://github.com/pingcap/tidb/pull/28670)

> 

Reviewer 6: - 修复配置项 `plugin_dir` 的默认值错误的问题 [28084](https://github.com/pingcap/tidb/issues/28084)

+ TiKV

    - 修复 Region 迁移时 Raftstore 模块出现死锁导致 TiKV 不可用的问题。用户可通过关闭调度并重启出问题的 TiKV 来临时应对。[#10909](https://github.com/tikv/tikv/issues/10909)
    - 修复因 Congest 错误而导致的 CDC 频繁增加 scan 重试的问题 [#11082](https://github.com/tikv/tikv/issues/11082)
    - 修复因 channel 打满而导致的 Raft 断连情况 [#11047](https://github.com/tikv/tikv/issues/11047)
    - 修复 Raft client 中 batch 消息过大的问题 [#9714](https://github.com/tikv/tikv/issues/9714)
    - 修复 `resolved_ts` 中协程泄漏的问题 [#10965](https://github.com/tikv/tikv/issues/10965)
    - 修复当 response 大小超过 4 GiB 时 Coprocessor panic 的问题 [#9012](https://github.com/tikv/tikv/issues/9012)
    - 修复当一个 snapshot 文件无法被垃圾清理 (GC) 时 snapshot GC 会缺失 GC snapshot 文件的问题 [#10813](https://github.com/tikv/tikv/issues/10813)
    - 修复当处理 Coprocessor 请求时因超时而导致 panic 的问题 [#10852](https://github.com/tikv/tikv/issues/10852)

> 

Reviewer 8: ```suggestion
    - 修复因统计线程监控数据导致的内存泄漏 [#11195](https://github.com/tikv/tikv/issues/11195)
    - 修复在某些平台获取 cgroup 信息导致 panic 的问题 [#10980](https://github.com/tikv/tikv/pull/10980) 
```

    > **注意：**
    >
    > 在 TSO 请求负载不高的情况下，不建议调整该参数。 

### 稳定性

- **支持 Raft 多数副本丢失下的在线恢复能力**

    增加 PD-CTL RECOVER-STORE-FAILURE 语句支持，提供在线恢复 TiKV 实例能力。通过该功能，用户可以实现：

    - 在线恢复所有 Raft 多数副本丢失的 Region 可读写。
    - 保证恢复所有 Region 后没有数据空洞。

    需要注意的是，Raft 多数失败的情况下无法避免已提交数据的丢失。

    [用户文档](/)  

### 高可用和容灾

- **功能 9**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

### 数据迁移

- **支持部署多个 TiDB Lightning**

    新版本 TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。 该功能无需特别的配置，在不改变用户使用习惯的同时，极大提高了用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。

    在产品性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。（数据待更新）。此外在 MySQL 分表数据聚合迁移到 TiDB 的场景中，MySQL 分表之间可能有冲突数据（主键/唯一键索引相同的数据），Lightning 也支持了数据导入过程中检查冲突数据的功能，用户可以使用该功能发现冲突数据，然后按照业务规则进行处理，冲突检测使用文档（待更新）。

    [用户文档](/)  

- **提高 DM 复制性能**

    支持以下功能，实现以更低的延迟将数据从 MySQL 同步数据到 TiDB。

    - 合并单行数据的多次变更（Compact multiple updates on a single row into one statement）
    - 点查更新合并为批量操作（Merge batch updates of multiple rows into one statement）
    - 异步保存检查点（Async Checkpoint）

> - **增加 DM 的 OpenAPI 以更方便地管理集群**

Reviewer 2: （实验特性）

    <功能描述 （DM 的 OpenAPI 是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

- **Lightning 支持导入 GBK 编码文件** 

    [用户文档](/tidb-lightning-configuration.md)

> - **Lightning 支持忽略部分错误行**

Reviewer 2: （实验特性）

    <能给用户带来什么好处>

> 

Reviewer 9: ```suggestion
- **Sync-diff-inspector 优化**

    - 大幅提升了对比速度，由原来的 375 MB/s 提升至 700 MB/s
    - 对比过程中对 TiDB 节点的内存消耗降低近一半
    - 优化了用户交互界面，在对比过程中可以显示进度
   
     [用户文档](/)
```

### TiDB 数据共享订阅

> - **功能 10**
> 
>     <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>
> 
>     <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>
> 
>     <如果功能限制或此功能特定的兼容性问题，需要提及>
> 
>     [用户文档](/)  

Reviewer 9: ```suggestion
- **TiCDC 支持灾备场景下的最终一致性复制**

    - 在主从灾备架构下，若上游集群遭遇不可恢复的灾难，TiCDC 具备保证下游最终一致性的能力。

    [用户文档](/)  
```

    - 改进 `tidb_analyze_version=2` 时 ANALYZE 默认的收集逻辑，提高收集速度并且降低资源开销
    - 引入语法 `ANALYZE TABLE table_name COLUMNS col_1, col_2, ..., col_n`，为宽表提供只收集一部分列统计信息的方案。提高宽表收集统计信息的速度。

+ TiKV

    - 增强磁盘空间防护能力，提升存储稳定性。

        针对 TiKV 遇到磁盘写满错误时可能 Panic 的问题，为磁盘剩余空间引入两级阈值防御机制，避免超额流量耗尽磁盘空间。同时，提供阈值触发时的空间回收能力。当剩余空间触发阈值时，部分写操作会失败，并返回 disk full 错误和盘满节点列表。此时，可以通过 `Drop/Truncate Table` 或者扩容等方式来回收空间，恢复服务。

    - 简化 L0 层流控算法 [#10879](https://github.com/tikv/tikv/issues/10879)
    - 优化 raft client 错误日志的收集 [#10944](https://github.com/tikv/tikv/pull/10944)
    - 优化日志线程以避免其成为性能瓶颈 [#10841](https://github.com/tikv/tikv/issues/10841)
    - 添加更多的写入查询统计类型 [#10507](https://github.com/tikv/tikv/issues/10507)

+ PD

    - 热点调度器的 QPS 维度支持更多的写请求类型 [#3869](https://github.com/tikv/pd/issues/3869)
    - 通过动态调整 Balance Region 调度器的重试上限，优化该调度器的性能 [#3744](https://github.com/tikv/pd/issues/3744)
    - 将 TiDB Dashboard 升级至 v2021.10.08.1 [#4070](https://github.com/tikv/pd/pull/4070)
    - 允许 Evict Leader 调度器调度拥有不健康副本的 Region [#4093](https://github.com/tikv/pd/issues/4093)
    - 优化调度器退出的速度 [#4146](https://github.com/tikv/pd/issues/4146)

+ TiFlash

>     - 显著优化了 TableScan 算子的执行效率

Reviewer: - 优化了 Exchange 算子的执行效率

Author: Done

---
title: TiDB 5.3 Release Notes
---

# TiDB 5.3 Release Notes

发版日期：2021 年 11 月 16 日

TiDB 版本：5.3.0

在 5.3 版本中，你可以获得以下关键特性：

+ 
+ 
+ 

## 兼容性变化

> **注意：**
>
> 当从一个早期的 TiDB 版本升级到 TiDB 5.3.0 时，如需了解所有中间版本对应的兼容性更改说明，请查看对应版本的 [Release Note](/releases/release-notes.md)。

### 系统变量

|  变量名    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- |
|tidb_enable_noop_functions  | 修改 | 此变量的控制范围不再包括 `CREATE TEMPORARY TABLE` 和 `DROP TEMPORARY TABLE` 行为。 |
| `tmp_table_size` | 修改 | 更名为 `tidb_tmp_table_size`，不再保留 `tmp_table_size`。升级前global 级别tmp_table_size 有值，升级后需要手动转换为tidb_tmp_table_size。 |
| tidb_tmp_table_max_size | 新增  | 此变量用于限制单个[临时表](/temporary-table.md)的最大大小，临时表超出该大小后报错。 |
|  |  |  |

### 配置文件参数

> |  配置文件    |  配置项    |  修改类型    |  描述    |
> | :---------- | :----------- | :----------- | :----------- |
> | TiDB |  |  |
> | TiKV | storage.reserve-space  | 原磁盘占位符 `storage.reserve-space` 会按功能进行拆分，其中的 20% 为磁盘文件用作硬防御，80% 用作软防御，空间回收。 |
> |  | memory-usage-limit  | 以前的版本没有 memory-usage-limit 参数， 升级后改参数值根据 storage.block-cache.capacity来计算 |
> | PD | max-snapshot-count |  |
> | PD | max-pending-peer-count |  |
> | PD | patrol-region-interval |  |
> | TiDB | prepared-plan-cache.capacity  | Increase default value for prepared-plan-cache.capacity from 100 to 1000 |
> |  |  |  |

Reviewer 9: ```suggestion
|  配置文件    |  配置项    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- | :----------- |
| TiDB |  |  |
| TiKV | storage.reserve-space  | 原磁盘占位符 `storage.reserve-space` 会按功能进行拆分，其中的 20% 为磁盘文件用作硬防御，80% 用作软防御，空间回收。 |
|  | memory-usage-limit  | 以前的版本没有 memory-usage-limit 参数， 升级后改参数值根据 storage.block-cache.capacity来计算 |
| PD | max-snapshot-count |  |
| PD | max-pending-peer-count |  |
| PD | patrol-region-interval |  |
| TiDB | prepared-plan-cache.capacity  | Increase default value for prepared-plan-cache.capacity from 100 to 1000 |
| TiCDC | consistent.level | 新增 | TiCDC 一致性复制级别 |
| TiCDC | consistent.max-log-size | 新增 | TiCDC redo log 存储单文件最大尺寸 |
| TiCDC | consistent.flush-interval | 新增 | TiCDC 刷新 redo log 的频率 |
| TiCDC | consistent.storage | 新增 | TiCDC 存储 redo  log 的位置 |

|  |  |  |
```

Author: 依据 https://github.com/pingcap/docs-cn/pull/7376/files， 这些是需要用户手动去增加一些配置开启灾难场景下的最终一致性复制特性，不是系统自己加的配置参数，所以不需要写到参数变化哪个表格里

    优化 TiDB 时间戳处理流程，支持通过开启 PD follower proxy 和调整 PD client 批量请求 TSO 的最大等待时长，降低 PD leader 时间戳处理负载，提升系统整体可扩展性。

    - 支持通过 tidb_enable_tso_follower_proxy 参数设置 PD follower proxy 功能开关。在 PD 时间戳请求负载过高的情况下，通过开启 PD follower proxy，可以将 follower 上请求周期内收集到的 TSO request 批量转发到 leader 节点，有效减少 client 与 leader 的直接交互次数，降低 leader 的负载，提升 TiDB 整体性能。

    > **注意：**
    >
    > 在 client 数较少的情况下，PD leader 负载不高的情况下，不建议开启 PD follower proxy 功能。

    - 支持通过 tidb_tso_client_batch_max_wait_time 参数设置 PD client 批量请求 TSO 的最大等待时间，单位毫秒。在 PD TSO 请求负载过高的情况下，通过调大该参数，可以提升一次请求 TSO 的数量，降低 PD 负载，提升吞吐。

    > **注意：**
    >
    > 在 TSO 请求负载不高的情况下，不建议调整该参数。 

### 稳定性

- **支持 Raft 多数副本丢失下的在线恢复能力**

    增加 PD-CTL RECOVER-STORE-FAILURE 语句支持，提供在线恢复 TiKV 实例能力。通过该功能，用户可以实现：

    - 在线恢复所有 Raft 多数副本丢失的 Region 可读写。
    - 保证恢复所有 Region 后没有数据空洞。

    需要注意的是，Raft 多数失败的情况下无法避免已提交数据的丢失。

    [用户文档](/)  

### 高可用和容灾

- **功能 9**

    <功能描述 （功能是什么 + 能给用户带来什么好处 + 需要用户注意什么）>

    <功能支持情况，TiDB 默认开启还是关闭此功能，如果默认关闭，如何开启>

    <如果功能限制或此功能特定的兼容性问题，需要提及>

    [用户文档](/)  

### 数据迁移

- **支持部署多个 TiDB Lightning**

    新版本 TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。 该功能无需特别的配置，在不改变用户使用习惯的同时，极大提高了用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。

>     在产品性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。（数据待更新）。此外在 MySQL 分表数据聚合迁移到 TiDB 的场景中，MySQL 分表之间可能有冲突数据（主键/唯一键索引相同的数据），Lightning 也支持了数据导入过程中检查冲突数据的功能，用户可以使用该功能发现冲突数据，然后按照业务规则进行处理，冲突检测使用文档（待更新）。

Reviewer 4: ```suggestion
    在产品性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。（数据待更新）。
```

Reviewer 4: TiDB Lightning  Data Import pre-check and duplication detect 这个功能目前经过测试，不满足 ga 标准，先不 ga

Author: 不 GA 的部分可以在 release notes 提到这是实验特性，这样就不用整句删除了是吗

    建议为访问 TiDB Dashboard 创建一个最小权限的 SQL 用户，并用该用户登录 TiDB Dashboard，避免使用高权限用户，提升安全性。

    [用户文档](/dashboard/dashboard-user.md)

### 性能

- **优化 PD 时间戳处理流程**

    优化 TiDB 时间戳处理流程，支持通过开启 PD Follower Proxy 和调整 PD client 批量请求 TSO 时所需的 batch 等待时间的方式来降低 PD leader 时间戳处理负载，提升系统整体可扩展性。

    - 支持通过系统变量 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) 设置 PD Follower Proxy 功能开关。在 PD 时间戳请求负载过高的情况下，通过开启 PD Follower Proxy，可以将 follower 上请求周期内收集到的 TSO request 批量转发到 leader 节点，从而有效减少 client 与 leader 的直接交互次数，降低 leader 的负载，提升 TiDB 整体性能。

    > **注意：**
    >
    > 在 client 数较少、PD leader 负载不高的情况下，不建议开启 PD Follower Proxy 功能。

    - 支持通过 [`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) 系统变量设置 PD client 批量请求 TSO 时所需的最大 batch 等待时间，单位为毫秒。在 PD TSO 请求负载过高的情况下，可以通过调大等待时间获得更大的 batch size，从而降低 PD 负载，提升吞吐。

    > **注意：**
    >
    > 在 TSO 请求负载不高的情况下，不建议调整该参数。

    [用户文档](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入)，[#3149](https://github.com/tikv/pd/issues/3149)

### 稳定性

- **支持多节点永久损坏后的在线有损恢复（实验特性）**

>     新增对 `unsafe remove-failed-stores` 命令的支持，实现数据有损恢复。当多数副本发生了永久性损坏（如磁盘损坏）等问题，导致无法在业务层读写一段数据时，PD 可以执行在线数据恢复，使该数据恢复至可读写状态。
> 
> 进行该功能相关操作时，建议在 TiDB 团队支持下完成。

Reviewer 10: ```suggestion
    新增对 `unsafe remove-failed-stores` 命令的支持，实现数据有损恢复。当多数副本发生了永久性损坏（如磁盘损坏）等问题，导致无法在业务层读写一段数据时，PD 可以执行在线数据恢复，使该数据恢复至可读写状态。

    进行该功能相关操作时，建议在 TiDB 团队支持下完成。
```

### 系统变量

|  变量名    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- |
| [tidb_enable_noop_functions](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) | 修改 | 此变量的控制范围不再包括 `CREATE TEMPORARY TABLE` 和 `DROP TEMPORARY TABLE` 行为。 |
| [`tidb_enable_pseudo_for_outdated_stats`](/system-variables.md#tidb_enable_pseudo_for_outdated_stats-从-v530-版本开始引入) | 新增 | 此变量用于控制优化器在一张表上的统计信息过期时的行为。默认值为 `ON`，当表数据被修改的行数大于该表总行数的 80% （该比例可通过 [`pseudo-estimate-ratio`](/tidb-configuration-file.md#pseudo-estimate-ratio) 配置项调整） 时，优化器认为该表上除总行数以外的统计信息不再可靠，转而使用 pseudo 统计信息。将该变量值设为 `OFF` 后，即使统计信息过期，优化器也仍会使用该表上的统计信息。|
|[`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) | 新增  | 此变量用于开启或关闭 TSO Follower Proxy 特性。默认值为 `OFF`，代表关闭TSO Follower Proxy 特性。此时，TiDB 仅会从 PD leader 获取 TSO。当开启该特性之后，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而降低 PD leader 的 CPU 压力。 |
|[`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) | 新增 | 此变量用于设置 TiDB 向 PD 请求 TSO 时进行一次攒批操作的最大等待时长。默认值为 `0`，即不进行额外的等待。 |

> | [tidb_tmp_table_max_size](/system-variables.md#tidb_tmp_table_max_size-从-v53-版本开始引入) | 新增  | 此变量用于限制单个[临时表](/temporary-table.md)的最大大小，临时表超出该大小后报错。 |

Reviewer 11: ```suggestion
| [tidb_tmp_table_max_size](/system-variables.md#tidb_tmp_table_max_size-从-v53-版本开始引入) | 新增  | 此变量用于限制单个[临时表](/temporary-tables.md)的最大大小，临时表超出该大小后报错。 |
```

### 配置文件参数

|  配置文件    |  配置项    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- | :----------- |
| TiDB | [`prepared-plan-cache.capacity`](/tidb-configuration-file.md#capacity)  | 修改 | 此配置项用于控制缓存语句的数量。默认值从 `100` 修改为 `1000`。 |
| TiKV | [`storage.reserve-space`](/tikv-configuration-file.md#reserve-space) | 修改 | 此配置项用于控制 TiKV 启动时用于保护磁盘的预留空间。从 v5.3.0 起，预留空间的 80% 用作磁盘空间不足时运维操作所需要的额外磁盘空间，剩余的 20% 为磁盘临时文件。 |

> | TiKV | memory-usage-limit | 修改  | 以前的版本没有 memory-usage-limit 参数， 升级后改参数值根据 storage.block-cache.capacity来计算。 |

Reviewer 12: ```suggestion
| TiKV | memory-usage-limit | 修改  | 以前的版本没有 memory-usage-limit 参数， 升级后该参数值根据 storage.block-cache.capacity来计算。 |
```

Reviewer 12: “升级后”，具体指的是什么场景呢？

| PD | [`log.file.max-days`](/pd-configuration-file.md#max-days) | 修改 | 此配置项用于控制日志保留的最长天数。默认值从 `1` 修改为 `0`。 |
| PD | [`log.file.max-backups`](/pd-configuration-file.md#max-backups) | 修改 | 此配置项用于控制日志文件保留的最大个数。默认值从 `7` 修改为 `0`。 |
| PD | [`patrol-region-interval`](/pd-configuration-file.md#patrol-region-interval) | 修改 |此配置项用于控制 replicaChecker 检查 Region 健康状态的运行频率，越短则运行越快，通常状况不需要调整。默认值从 `100ms` 修改为 `10ms`。 |
| PD | [`max-snapshot-count`](/pd-configuration-file.md#max-snapshot-count) | 修改 |此配置项用于控制单个 store 最多同时接收或发送的 snapshot 数量，调度受制于这个配置来防止抢占正常业务的资源。默认值从 `3` 修改为 `64`。 |
| PD | [`max-pending-peer-count`](/pd-configuration-file.md#max-pending-peer-count) | 修改 |此配置项用于控制单个 store 的 pending peer 上限，调度受制于这个配置来防止在部分节点产生大量日志落后的 Region。默认值从 `16` 修改为 `64`。 |

### 其他

- 临时表：

>     - 如果在 v5.3.0 升级前创建了本地临时表，这些临时表实际为普通表，在升级后也会被 TiDB 当成普通表处理。在 v5.3.0 上创建的全局临时表在降级后会被当作普通表处理，导致数据错误。

Reviewer 10: ```suggestion
    - 对于本地临时表，如果在 v5.3.0 升级前创建了本地临时表，这些临时表实际为普通表，在升级后也会被 TiDB 当成普通表处理。对于全局临时表，如果在 v5.3.0 上创建了全局临时表，当 TiDB 降级后，这些临时表会被当作普通表处理，导致数据错误。
```

>     - TiCDC 和 BR 从 v5.3.0 开始支持[全局临时表](/temporary-table.md#全局临时表)。如果使用 v5.3.0 以下版本同步全局临时表到下游，会导致表定义错误。

Reviewer 11: ```suggestion
    - TiCDC 和 BR 从 v5.3.0 开始支持[全局临时表](/temporary-tables.md#全局临时表)。如果使用 v5.3.0 以下版本同步全局临时表到下游，会导致表定义错误。
```

    - 通过 TiDB 生态工具导入的集群、恢复后的集群、同步的下游集群必须是 TiDB v5.3.0 及以上版本，否则创建全局临时表时报错。

>     - 关于临时表的更多兼容性信息，请参考 [与 MySQL 临时表的兼容性](/temporary-table.md#与-mysql-临时表的兼容性)和[与其他 TiDB 功能的兼容性限制](/temporary-table.md#与其他-tidb-功能的兼容性限制)。

Reviewer 11: ```suggestion
    - 关于临时表的更多兼容性信息，请参考 [与 MySQL 临时表的兼容性](/temporary-tables.md#与-mysql-临时表的兼容性)和[与其他 TiDB 功能的兼容性限制](/temporary-table.md#与其他-tidb-功能的兼容性限制)。
```

- 修正 `SHOW CREATE VIEW` 不需要 `SHOW VIEW` 权限的问题，现在用户必须具有 `SHOW VIEW` 权限才允许执行 `SHOW CREATE VIEW` 语句。
- 系统变量 `sql_auto_is_null` 被加入 Noop Function 中，当 `tidb_enable_noop_functions = 0/OFF` 时，修改该变量会报错。
- 不再允许执行 `GRANT ALL ON performance_schema.*` 语法，在 TiDB 上执行该语句会报错。

> - v5.3.0 之前，对于新增索引，analyze 时间不受设定时间的限制，`tidb_auto_analyze_start_time` 和 `tidb_auto_analyze_end_time` 时间段内将不会触发 auto analyze
> - plugin 的默认路径从 "" 改为 /data/deploy/plugin
> - DM 代码迁移至 [TiCDC 代码仓库的 dm 文件夹](https://github.com/pingcap/ticdc/tree/master/dm)。DM 版本号从 v2.0.x 修改为 v5.3.0，用户可以无风险从 v2.0.x 升级至 v5.3。

Reviewer 10: ```suggestion
- v5.3.0 之前，对于新增索引，analyze 时间不受设定时间的限制，`tidb_auto_analyze_start_time` 和 `tidb_auto_analyze_end_time` 时间段内将不会触发 auto analyze。
- plugin 的默认路径从 "" 改为 /data/deploy/plugin。
- DM 代码迁移至 [TiCDC 代码仓库的 dm 文件夹]。(https://github.com/pingcap/ticdc/tree/master/dm)。DM 版本号从 v2.0.x 修改为 v5.3.0，用户可以无风险从 v2.0.x 升级至 v5.3。
```

## 新功能

### SQL

- **使用 SQL 接口设置数据放置规则（实验特性）**

    新增对 `[CREATE | ALTER] PLACEMENT POLICY` 语句的支持，提供 SQL 接口设置数据放置规则。通过该功能，用户可以指定任意连续数据按照不同地域、机房、机柜、主机、硬件、副本数规则进行部署，满足低成本、高可用、灵活多变的业务诉求。该功能可以实现以下业务场景：

>     - 跨区域放置数据以改善局部访问性能
>     - 合并多个不同业务的数据库，大幅减少数据库常规运维管理的成本，并通过规则配置实现业务资源隔离
>     - 增加重要数据的副本数，提高业务可用性和数据可靠性
>     - 将最新数据存入 SSD，历史数据存入 HDD，降低归档数据存储成本
>     - 把热点数据的 leader 放到高性能的 TiKV 实例上
>     - 将不相关的数据分离到不同的存储中以提高可用性
> 

Reviewer 11: 根据 https://github.com/pingcap/docs/pull/6766#discussion_r746788710 中的讨论，删除第一条用户场景。
```suggestion
    - 合并多个不同业务的数据库，大幅减少数据库常规运维管理的成本，并通过规则配置实现业务资源隔离
    - 增加重要数据的副本数，提高业务可用性和数据可靠性
    - 将最新数据存入 SSD，历史数据存入 HDD，降低归档数据存储成本
    - 把热点数据的 leader 放到高性能的 TiKV 实例上
    - 将冷数据分离到不同的存储中以提高可用性

```

    [用户文档](/information-schema/information-schema-placement-rules.md)，[#18030](https://github.com/pingcap/tidb/issues/18030)

- **临时表**

    新增对 `CREATE [GLOBAL] TEMPORARY TABLE` 语句的支持。支持创建临时表，方便管理业务中间计算的临时数据。临时表中的数据均保存在内存中，用户可通过 `tidb_tmp_table_max_size` 变量限制临时表的内存大小。TiDB 支持以下两种临时表：

    - Global 临时表
        - 对集群内所有 session 可见，表结构持久化。
        - 提供事务级别的数据隔离，数据只在事务内有效，事务结束后自动删除数据。
    - Local 临时表
        - 只对当前 session 可见，表结构不持久化。
        - 支持重名，用户无需为业务设计复杂的表命名规则。
        - 提供会话级别的数据隔离，降低业务设计复杂度，会话结束后删除临时表。

    [用户文档](/temporary-table.md)，[#24169](https://github.com/pingcap/tidb/issues/24169)

- **支持 `FOR UPDATE OF TABLES` 语法**

    对于存在多表 join 的语句，支持只对 `OF TABLES` 中包含的表关联的行进行悲观锁加锁操作。

    [用户文档](/sql-statements/sql-statement-select.md)，[#28689](https://github.com/pingcap/tidb/issues/28689)

> - **表属性设置**
> 
>     增加 `ALTER TABLE [PARTITION] ATTRIBUTES` 语句支持，允许用户设置表和分区的表属性。目前支持设置 `merge_option` 属性。通过设置 `merge_option` 属性，用户可以显式控制 Region 是否合并。
> 
>     应用场景：当用户 `SPLIT TABLE` 之后，如果超过一定时间后没有插入数据，空 Region 默认会被自动合并。此时，可以通过该功能设置表属性为 `merge_option=deny`，避免 Region 的自动合并。

Reviewer 10: ```suggestion
- **设置表属性**

    增加 `ALTER TABLE [PARTITION] ATTRIBUTES` 语句支持，允许用户设置表和分区的表属性。目前 TiDB 仅支持设置 `merge_option` 属性。通过为表或分区添加 `merge_option` 属性，用户可以显式控制 Region 是否合并。

    应用场景：当用户 `SPLIT TABLE` 之后，如果超过一定时间后没有插入数据，空 Region 默认会被自动合并。此时，可以通过该功能设置表属性为 `merge_option=deny`，避免 Region 的自动合并。
```

    [用户文档](/table-attributes.md)，[#3839](https://github.com/tikv/pd/issues/3839)

### 安全

- **支持为 TiDB Dashboard 创建最小权限用户**

    TiDB Dashboard 的账号体系与 TiDB SQL 用户一致，并基于 TiDB SQL 用户的权限进行 TiDB Dashboard 授权验证。TiDB Dashboard 所需的权限较少，甚至可以只有只读权限。可以基于最小权限原则配置合适的用户访问 TiDB Dashboard，减少高权限用户的使用场景。

    建议为访问 TiDB Dashboard 创建一个最小权限的 SQL 用户，并用该用户登录 TiDB Dashboard，避免使用高权限用户，提升安全性。

    [用户文档](/dashboard/dashboard-user.md)

### 性能

- **优化 PD 时间戳处理流程**

>     优化 TiDB 时间戳处理流程，支持通过开启 PD follower proxy 和调整 PD client 批量请求 TSO 的最大等待时长，降低 PD leader 时间戳处理负载，提升系统整体可扩展性。
> 
>     - 支持通过 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) 系统变量设置 PD follower proxy 功能开关。在 PD 时间戳请求负载过高的情况下，通过开启 PD follower proxy，可以将 follower 上请求周期内收集到的 TSO request 批量转发到 leader 节点，有效减少 client 与 leader 的直接交互次数，降低 leader 的负载，提升 TiDB 整体性能。
> 
>     > **注意：**
>     >
>     > 在 client 数较少的情况下，PD leader 负载不高的情况下，不建议开启 PD follower proxy 功能。

Reviewer 10: ```suggestion
    优化 TiDB 时间戳处理流程，支持通过开启 PD follower proxy 和调整 PD client 批量请求 TSO 的最大等待时长的方式降低 PD leader 时间戳处理负载，提升系统整体可扩展性。

    - 支持通过系统变量 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) 设置 PD follower proxy 功能开关。在 PD 时间戳请求负载过高的情况下，通过开启 PD follower proxy，可以将 follower 上请求周期内收集到的 TSO request 批量转发到 leader 节点，从而有效减少 client 与 leader 的直接交互次数，降低 leader 的负载，提升 TiDB 整体性能。

    > **注意：**
    >
    > 在 client 数较少、PD leader 负载不高的情况下，不建议开启 PD follower proxy 功能。
```

>     优化 TiDB 时间戳处理流程，支持通过开启 PD follower proxy 和调整 PD client 批量请求 TSO 的最大等待时长，降低 PD leader 时间戳处理负载，提升系统整体可扩展性。
> 
>     - 支持通过 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) 系统变量设置 PD follower proxy 功能开关。在 PD 时间戳请求负载过高的情况下，通过开启 PD follower proxy，可以将 follower 上请求周期内收集到的 TSO request 批量转发到 leader 节点，有效减少 client 与 leader 的直接交互次数，降低 leader 的负载，提升 TiDB 整体性能。
> 
>     > **注意：**
>     >
>     > 在 client 数较少的情况下，PD leader 负载不高的情况下，不建议开启 PD follower proxy 功能。
> 
>     - 支持通过 [`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) 系统变量设置 PD client 批量请求 TSO 的最大等待时间，单位毫秒。在 PD TSO 请求负载过高的情况下，通过调大该参数，可以提升一次请求 TSO 的数量，降低 PD 负载，提升吞吐。
> 
>     > **注意：**
>     >
>     > 在 TSO 请求负载不高的情况下，不建议调整该参数。

Reviewer 10: ```suggestion
    优化 TiDB 时间戳处理流程，支持通过开启 PD follower proxy 和调整 PD client 批量请求 TSO 时所需的 batch 等待时间的方式来降低 PD leader 时间戳处理负载，提升系统整体可扩展性。
    - 支持通过系统变量 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) 设置 PD follower proxy 功能开关。在 PD 时间戳请求负载过高的情况下，通过开启 PD follower proxy，可以将 follower 上请求周期内收集到的 TSO request 批量转发到 leader 节点，从而有效减少 client 与 leader 的直接交互次数，降低 leader 的负载，提升 TiDB 整体性能。
    > **注意：**
    >
    > 在 client 数较少、PD leader 负载不高的情况下，不建议开启 PD follower proxy 功能。

    - 支持通过 [`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) 系统变量设置 PD client 批量请求 TSO 时所需的最大 batch 等待时间，单位为毫秒。在 PD TSO 请求负载过高的情况下，可以通过调大等待时间获得更大的 batch size，从而降低 PD 负载，提升吞吐。

    > **注意：**
    >
    > 在 TSO 请求负载不高的情况下，不建议调整该参数。
```

    [用户文档](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入)，[#3149](https://github.com/tikv/pd/issues/3149)

### 稳定性

> - **支持 Raft 多数副本丢失时数据的在线恢复能力（实验特性）**
> 
>     增加 PD-CTL RECOVER-STORE-FAILURE 语句支持，提供在线恢复 TiKV 实例能力。通过该功能，可以实现：
> 
>     - 在线恢复所有 Raft 多数副本丢失的 Region 可读写。
>     - 保证恢复所有 Region 后没有数据空洞。
> 
>     需要注意的是，Raft 多数副本失败的情况下无法避免已提交数据的丢失。

Reviewer 10: ```suggestion
- **支持多节点永久损坏后的在线有损恢复（实验特性）**

    增加 `unsafe remove-failed-stores` 相关命令支持，实现数据有损恢复。当多数副本发生了永久性损坏（如磁盘损坏）等问题，导致无法在业务层读写一段数据时，PD 可以执行在线执行数据恢复，使该数据恢复至可读写状态。

> **警告：**
>
> - 此功能为有损恢复，无法保证数据和数据索引完整性。
> - 此功能为实验特性，建议在生产环境中使用。
> - 建议在 TiDB 团队支持下进行相关操作，操作不当可能导致集群难以恢复。
```

>     [用户文档](/online-unsafe-recovery.md)，[#10483](https://github.com/tikv/tikv/issues/10483)

Reviewer 13: 1. 可以，LGTM
2. RECOVER-STORE-FAILURE 就是指的 unsafe remove-failed-stores，这里改下表述吧

### 数据迁移

- **提高 DM 同步性能**

    支持以下功能，实现以更低的延迟将数据从 MySQL 同步数据到 TiDB。

    - 合并单行数据的多次变更（Compact multiple updates on a single row into one statement）
    - 点查更新合并为批量操作（Merge batch updates of multiple rows into one statement）

- **增加 DM 的 OpenAPI 以更方便地管理集群（实验特性）**

    DM 提供 OpenAPI 功能，用户可通过 OpenAPI 对 DM 集群进行查询和运维操作。OpenAPI 的总体功能和 [dmctl 工具](https://docs.pingcap.com/zh/tidb-data-migration/stable/dmctl-introduction)类似。

    当前 OpenAPI 功能为实验特性，默认关闭，不建议在生产环境中使用。

    [用户文档](https://docs.pingcap.com/zh/tidb-data-migration/stable/open-api)

> - **TiDB Lightning 分布式并行导入**

Reviewer 2: ```suggestion
- **TiDB Lightning 并行导入**
```

    TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。该功能无需特别的配置，在不改变用户使用习惯的同时，极大提高了用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。

>     在产品性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。（数据待更新）。

Reviewer 14: ```suggestion
经测试，使用 10 台 TiDB Lightning，20 TB 规模的 MySQL 分表数据可以在  8 小时内导入到 TiDB，单台 TiDB Lightning 可以支持 250 GB/s 的导入速度，整体效率提升了 8 倍。

```

| PD | [`log.file.max-days`](/pd-configuration-file.md#max-days) | 修改 | 此配置项用于控制日志保留的最长天数。默认值从 `1` 修改为 `0`。 |
| PD | [`log.file.max-backups`](/pd-configuration-file.md#max-backups) | 修改 | 此配置项用于控制日志文件保留的最大个数。默认值从 `7` 修改为 `0`。 |
| PD | [`patrol-region-interval`](/pd-configuration-file.md#patrol-region-interval) | 修改 |此配置项用于控制 replicaChecker 检查 Region 健康状态的运行频率，越短则运行越快，通常状况不需要调整。默认值从 `100ms` 修改为 `10ms`。 |
| PD | [`max-snapshot-count`](/pd-configuration-file.md#max-snapshot-count) | 修改 |此配置项用于控制单个 store 最多同时接收或发送的 snapshot 数量，调度受制于这个配置来防止抢占正常业务的资源。默认值从 `3` 修改为 `64`。 |
| PD | [`max-pending-peer-count`](/pd-configuration-file.md#max-pending-peer-count) | 修改 |此配置项用于控制单个 store 的 pending peer 上限，调度受制于这个配置来防止在部分节点产生大量日志落后的 Region。默认值从 `16` 修改为 `64`。 |

### 其他

- 临时表：

    - 如果在 v5.3.0 升级前创建了本地临时表，这些临时表实际为普通表，在升级后也会被 TiDB 当成普通表处理。在 v5.3.0 上创建的全局临时表在降级后会被当作普通表处理，导致数据错误。
    - TiCDC 和 BR 从 v5.3.0 开始支持[全局临时表](/temporary-tables.md#全局临时表)。如果使用 v5.3.0 以下版本同步全局临时表到下游，会导致表定义错误。
    - 通过 TiDB 生态工具导入的集群、恢复后的集群、同步的下游集群必须是 TiDB v5.3.0 及以上版本，否则创建全局临时表时报错。
    - 关于临时表的更多兼容性信息，请参考 [与 MySQL 临时表的兼容性](/temporary-tables.md#与-mysql-临时表的兼容性)和[与其他 TiDB 功能的兼容性限制](/temporary-table.md#与其他-tidb-功能的兼容性限制)。

> - 修正 `SHOW CREATE VIEW` 不需要 `SHOW VIEW` 权限的问题，现在用户必须具有 `SHOW VIEW` 权限才允许执行 `SHOW CREATE VIEW` 语句。
> - 系统变量 `sql_auto_is_null` 被加入 Noop Function 中，当 `tidb_enable_noop_functions = 0/OFF` 时，修改该变量会报错。
> - 不再允许执行 `GRANT ALL ON performance_schema.*` 语法，在 TiDB 上执行该语句会报错。
> - 修复了 v5.3.0 之前的版本中新增索引会导致在规定时间外触发 auto-analyze 的问题。在 v5.3.0 中，用户通过 `tidb_auto_analyze_start_time` 和 `tidb_auto_analyze_end_time` 设定时间段后，只会在该时间段内触发 auto-analyze。

Reviewer 10: ```suggestion
- 修复 `SHOW CREATE VIEW` 不需要 `SHOW VIEW` 权限的问题，现在用户必须具有 `SHOW VIEW` 权限才允许执行 `SHOW CREATE VIEW` 语句。
- 系统变量 `sql_auto_is_null` 被加入 Noop Function 中，当 `tidb_enable_noop_functions = 0/OFF` 时，修改该变量会报错。
- 不再允许执行 `GRANT ALL ON performance_schema.*` 语法，在 TiDB 上执行该语句会报错。
- 修复 v5.3.0 之前的版本中新增索引会导致在规定时间外触发 auto-analyze 的问题。在 v5.3.0 中，用户通过 `tidb_auto_analyze_start_time` 和 `tidb_auto_analyze_end_time` 设定时间段后，只会在该时间段内触发 auto-analyze。
```

    [用户文档](/online-unsafe-recovery.md)，[#10483](https://github.com/tikv/tikv/issues/10483)

### 数据迁移

- **提高 DM 同步性能**

    支持以下功能，实现以更低的延迟将数据从 MySQL 同步数据到 TiDB。

    - 合并单行数据的多次变更（Compact multiple updates on a single row into one statement）
    - 点查更新合并为批量操作（Merge batch updates of multiple rows into one statement）

- **增加 DM 的 OpenAPI 以更方便地管理集群（实验特性）**

    DM 提供 OpenAPI 功能，用户可通过 OpenAPI 对 DM 集群进行查询和运维操作。OpenAPI 的总体功能和 [dmctl 工具](https://docs.pingcap.com/zh/tidb-data-migration/stable/dmctl-introduction)类似。

    当前 OpenAPI 功能为实验特性，默认关闭，不建议在生产环境中使用。

    [用户文档](https://docs.pingcap.com/zh/tidb-data-migration/stable/open-api)

- **TiDB Lightning 并行导入**

    TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。该功能无需特别的配置，在不改变用户使用习惯的同时，极大提高了用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。

    在产品性能测试中，使用 x 个 Lightning 导入整体大小 x TB MySQL 分表数据到 TiDB 单表，总耗时 x h，平均单台 Lightning 速度达到 x GB/h。（数据待更新）。

    [用户文档](/tidb-lightning/tidb-lightning-distributed-import.md)

- **TiDB Lightning 执行任务前的检查项**

    TiDB Lightning 增加了执行前检查配置的功能。默认开启。该功能会自动进行一些磁盘空间和执行配置的常规检查，主要目的是确保后续的整个导入过程顺利。

    [用户文档](tidb-lightning/tidb-lightning-prechecks.md)

- **TiDB Lightning 支持导入 GBK 编码的文件**

> 

Reviewer 15: ```suggestion
通过指定源数据文件的字符集，TiDB Lightning 会在导入过程中将源文件从指定的字符集转换为 UTF-8 编码。
```

    [用户文档](/dashboard/dashboard-user.md)

### 性能

- **优化 PD 时间戳处理流程**

    优化 TiDB 时间戳处理流程，支持通过开启 PD Follower Proxy 和调整 PD client 批量请求 TSO 时所需的 batch 等待时间的方式来降低 PD leader 时间戳处理负载，提升系统整体可扩展性。

    - 支持通过系统变量 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) 设置 PD Follower Proxy 功能开关。在 PD 时间戳请求负载过高的情况下，通过开启 PD Follower Proxy，可以将 follower 上请求周期内收集到的 TSO request 批量转发到 leader 节点，从而有效减少 client 与 leader 的直接交互次数，降低 leader 的负载，提升 TiDB 整体性能。

    > **注意：**
    >
    > 在 client 数较少、PD leader 负载不高的情况下，不建议开启 PD Follower Proxy 功能。

    - 支持通过 [`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) 系统变量设置 PD client 批量请求 TSO 时所需的最大 batch 等待时间，单位为毫秒。在 PD TSO 请求负载过高的情况下，可以通过调大等待时间获得更大的 batch size，从而降低 PD 负载，提升吞吐。

    > **注意：**
    >
    > 在 TSO 请求负载不高的情况下，不建议调整该参数。

    [用户文档](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入)，[#3149](https://github.com/tikv/pd/issues/3149)

### 稳定性

- **支持多节点永久损坏后的在线有损恢复（实验特性）**

    新增对 `unsafe remove-failed-stores` 命令的支持，实现数据有损恢复。当多数副本发生了永久性损坏（如磁盘损坏）等问题，导致无法在业务层读写一段数据时，PD 可以执行在线数据恢复，使该数据恢复至可读写状态。

> > **警告：**
> >
> > - 此功能为有损恢复，无法保证数据和数据索引完整性。
> > - 此功能为实验特性，其接口、策略和内部实现在 GA 前可能会有所变化。虽然已通过部分场景的测试，但尚未经过广泛验证，使用此功能可能导致系统不可用，不建议在生产环境中使用。
> > - 建议在 TiDB 团队支持下进行相关操作，操作不当可能导致集群难以恢复。

Reviewer 10: ```suggestion
进行该功能的相关操作时，建议在 TiDB 团队支持下完成。
```

---
title: TiDB 5.3 Release Notes
---

# TiDB 5.3 Release Notes

> 发版日期：2021 年 11 月 23 日

Reviewer 11: ```suggestion
发版日期：2021 年 11 月 30 日
```

TiDB 版本：5.3.0

在 5.3 版本中，你可以获得以下关键特性：

+ 引入临时表，简化业务逻辑并提升性能
+ 支持设置表和分区的表属性
+ 支持为 TiDB Dashboard 创建最小权限用户，提高系统安全性
+ 优化 TiDB 时间戳处理流程，提升系统的整体性能
+ 提高 DM 同步性能，实现以更低的延迟将数据从 MySQL 同步数据到 TiDB
+ 支持 TiDB Lightning 分布式并行导入，提升全量数据迁移效率
+ 支持“一键”保存和恢复现场问题的相关信息，提升查询计划问题诊断的效率
+ 支持持续性能分析 (Continuous Profiling) 实验特性，提高数据库性能的可观测性
+ 持续优化存储和计算引擎，提升系统性能和稳定性

## 兼容性变化

> **注意：**
>
> 当从一个早期的 TiDB 版本升级到 TiDB 5.3.0 时，如需了解所有中间版本对应的兼容性更改说明，请查看对应版本的 [Release Note](/releases/release-notes.md)。

### 系统变量

|  变量名    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- |

> | [tidb_enable_noop_functions](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) | 修改 | 此变量的控制范围不再包括 `CREATE TEMPORARY TABLE` 和 `DROP TEMPORARY TABLE` 行为。 |

Reviewer 15: ```suggestion
| [tidb_enable_noop_functions](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) | 修改 | 由于 TiDB 现在支持临时表，此变量的控制范围不再包括 `CREATE TEMPORARY TABLE` 和 `DROP TEMPORARY TABLE` 行为。 |
```

|[`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) | 新增  | 此变量用于开启或关闭 TSO Follower Proxy 特性。默认值为 `OFF`，代表关闭 TSO Follower Proxy 特性。此时，TiDB 仅会从 PD leader 获取 TSO。当开启该特性之后，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而降低 PD leader 的 CPU 压力。 |
|[`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) | 新增 | 此变量用于设置 TiDB 向 PD 请求 TSO 时进行一次攒批操作的最大等待时长。默认值为 `0`，即不进行额外的等待。 |
| [tidb_tmp_table_max_size](/system-variables.md#tidb_tmp_table_max_size-从-v53-版本开始引入) | 新增  | 此变量用于限制单个[临时表](/temporary-tables.md)的最大大小，临时表超出该大小后报错。 |

### 配置文件参数

|  配置文件    |  配置项    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- | :----------- |
| TiDB | [`prepared-plan-cache.capacity`](/tidb-configuration-file.md#capacity)  | 修改 | 此配置项用于控制缓存语句的数量。默认值从 `100` 修改为 `1000`。 |
| TiKV | [`storage.reserve-space`](/tikv-configuration-file.md#reserve-space) | 修改 | 此配置项用于控制 TiKV 启动时用于保护磁盘的预留空间。从 v5.3.0 起，预留空间的 80% 用作磁盘空间不足时运维操作所需要的额外磁盘空间，剩余的 20% 为磁盘临时文件。 |

> | TiKV | memory-usage-limit | 修改  | 以前的版本没有 memory-usage-limit 参数， 升级后该参数值根据 storage.block-cache.capacity 来计算。 |

Reviewer 11: ```suggestion
| TiKV | `memory-usage-limit` | 修改  | 以前的版本没有 memory-usage-limit 参数， 升级后该参数值根据 storage.block-cache.capacity 来计算。 |
|  TiKV | `raftstore.raft-reject-transfer-leader-duration` | 删除 | 控制迁移 leader 到新加节点的最小时间。|
```

---
title: TiDB 5.3 Release Notes
---

# TiDB 5.3 Release Notes

发版日期：2021 年 11 月 23 日

TiDB 版本：5.3.0

在 5.3 版本中，你可以获得以下关键特性：

+ 引入临时表，简化业务逻辑并提升性能
+ 支持设置表和分区的表属性
+ 支持为 TiDB Dashboard 创建最小权限用户，提高系统安全性
+ 优化 TiDB 时间戳处理流程，提升系统的整体性能

> + 提高 DM 复制性能，实现以更低的延迟将数据从 MySQL 同步数据到 TiDB

Reviewer 15: ```suggestion
+ 提高 DM 同步性能，实现以更低的延迟将数据从 MySQL 同步数据到 TiDB
```

| PD | [`log.file.max-days`](/pd-configuration-file.md#max-days) | 修改 | 此配置项用于控制日志保留的最长天数。默认值从 `1` 修改为 `0`。 |
| PD | [`log.file.max-backups`](/pd-configuration-file.md#max-backups) | 修改 | 此配置项用于控制日志文件保留的最大个数。默认值从 `7` 修改为 `0`。 |
| PD | [`patrol-region-interval`](/pd-configuration-file.md#patrol-region-interval) | 修改 | 此配置项用于控制 replicaChecker 检查 Region 健康状态的运行频率，越短则运行越快，通常状况不需要调整。默认值从 `100ms` 修改为 `10ms`。 |
| PD | [`max-snapshot-count`](/pd-configuration-file.md#max-snapshot-count) | 修改 | 此配置项用于控制单个 store 最多同时接收或发送的 snapshot 数量，调度受制于这个配置来防止抢占正常业务的资源。默认值从 `3` 修改为 `64`。 |
| PD | [`max-pending-peer-count`](/pd-configuration-file.md#max-pending-peer-count) | 修改 | 此配置项用于控制单个 store 的 pending peer 上限，调度受制于这个配置来防止在部分节点产生大量日志落后的 Region。默认值从 `16` 修改为 `64`。 |

### 其他

- 临时表：

    - 对于本地临时表，如果在 v5.3.0 升级前创建了本地临时表，这些临时表实际为普通表，在升级至 v5.3.0 或更高版本后，也会被 TiDB 当成普通表处理。对于全局临时表，如果在 v5.3.0 上创建了全局临时表，当 TiDB 降级至 v5.3.0 以前版本后，这些临时表会被当作普通表处理，导致数据错误。
    - TiCDC 和 BR 从 v5.3.0 开始支持[全局临时表](/temporary-tables.md#全局临时表)。如果使用 v5.3.0 以下版本同步全局临时表到下游，会导致表定义错误。
    - 通过 TiDB 生态工具导入的集群、恢复后的集群、同步的下游集群必须是 TiDB v5.3.0 及以上版本，否则创建全局临时表时报错。

>     - 关于临时表的更多兼容性信息，请参考 [与 MySQL 临时表的兼容性](/temporary-tables.md#与-mysql-临时表的兼容性)和[与其他 TiDB 功能的兼容性限制](/temporary-table.md#与其他-tidb-功能的兼容性限制)。

Reviewer 11: ```suggestion
    - 关于临时表的更多兼容性信息，请参考 [与 MySQL 临时表的兼容性](/temporary-tables.md#与-mysql-临时表的兼容性)和[与其他 TiDB 功能的兼容性限制](/temporary-tables.md#与其他-tidb-功能的兼容性限制)。
```

- plugin 默认存放目录从 "" 改为 /data/deploy/plugin。
- DM 代码迁移至 [TiCDC 代码仓库的 dm 文件夹](https://github.com/pingcap/ticdc/tree/master/dm)。DM 版本号从 v2.0.x 修改为 v5.3.0，用户可以无风险从 v2.0.x 升级至 v5.3.0。

## 新功能

### SQL

- **使用 SQL 接口设置数据放置规则（实验特性）**

    新增对 `[CREATE | ALTER] PLACEMENT POLICY` 语句的支持，提供 SQL 接口设置数据放置规则。通过该功能，用户可以指定任意连续数据按照不同地域、机房、机柜、主机、硬件、副本数规则进行部署，满足低成本、高可用、灵活多变的业务诉求。该功能可以实现以下业务场景：

    - 合并多个不同业务的数据库，大幅减少数据库常规运维管理的成本，并通过规则配置实现业务资源隔离
    - 增加重要数据的副本数，提高业务可用性和数据可靠性
    - 将最新数据存入 SSD，历史数据存入 HDD，降低归档数据存储成本
    - 把热点数据的 leader 放到高性能的 TiKV 实例上
    - 将冷数据分离到不同的存储中以提高可用性

    [用户文档](/information-schema/information-schema-placement-rules.md)，[#18030](https://github.com/pingcap/tidb/issues/18030)

- **临时表**

    新增对 `CREATE [GLOBAL] TEMPORARY TABLE` 语句的支持。支持创建临时表，方便管理业务中间计算的临时数据。临时表中的数据均保存在内存中，用户可通过 `tidb_tmp_table_max_size` 变量限制临时表的内存大小。TiDB 支持以下两种临时表：

    - Global 临时表
        - 对集群内所有 session 可见，表结构持久化。
        - 提供事务级别的数据隔离，数据只在事务内有效，事务结束后自动删除数据。
    - Local 临时表
        - 只对当前 session 可见，表结构不持久化。
        - 支持重名，用户无需为业务设计复杂的表命名规则。
        - 提供会话级别的数据隔离，降低业务设计复杂度，会话结束后删除临时表。

>     [用户文档](/temporary-table.md)，[#24169](https://github.com/pingcap/tidb/issues/24169)

Reviewer 11: ```suggestion
    [用户文档](/temporary-tables.md)，[#24169](https://github.com/pingcap/tidb/issues/24169)
```

    进行该功能相关操作时，建议在 TiDB 团队支持下完成。

    [用户文档](/online-unsafe-recovery.md)，[#10483](https://github.com/tikv/tikv/issues/10483)

### 数据迁移

- **提高 DM 同步性能**

    支持以下功能，实现以更低的延迟将数据从 MySQL 同步数据到 TiDB。

    - 合并单行数据的多次变更（Compact multiple updates on a single row into one statement）
    - 点查更新合并为批量操作（Merge batch updates of multiple rows into one statement）

- **增加 DM 的 OpenAPI 以更方便地管理集群（实验特性）**

    DM 提供 OpenAPI 功能，用户可通过 OpenAPI 对 DM 集群进行查询和运维操作。OpenAPI 的总体功能和 [dmctl 工具](https://docs.pingcap.com/zh/tidb-data-migration/stable/dmctl-introduction)类似。

    当前 OpenAPI 功能为实验特性，默认关闭，不建议在生产环境中使用。

    [用户文档](https://docs.pingcap.com/zh/tidb-data-migration/stable/open-api)

- **TiDB Lightning 并行导入**

    TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。该功能无需特别的配置，在不改变用户使用习惯的同时，极大提高了用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。

经测试，使用 10 台 TiDB Lightning，20 TB 规模的 MySQL 数据可以在 8 小时内导入到 TiDB，单台 TiDB Lightning 可以支持 250 GB/s 的导入速度，整体效率提升了 8 倍。

    [用户文档](/tidb-lightning/tidb-lightning-distributed-import.md)

- **TiDB Lightning 执行任务前的检查项**

    TiDB Lightning 增加了执行前检查配置的功能。默认开启。该功能会自动进行一些磁盘空间和执行配置的常规检查，主要目的是确保后续的整个导入过程顺利。

>     [用户文档](tidb-lightning/tidb-lightning-prechecks.md)

Reviewer 11: ```suggestion
    [用户文档](/tidb-lightning/tidb-lightning-prechecks.md)
```

|[`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) | 新增  | 此变量用于开启或关闭 TSO Follower Proxy 特性。默认值为 `OFF`，代表关闭 TSO Follower Proxy 特性。此时，TiDB 仅会从 PD leader 获取 TSO。当开启该特性之后，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而降低 PD leader 的 CPU 压力。 |
|[`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) | 新增 | 此变量用于设置 TiDB 向 PD 请求 TSO 时进行一次攒批操作的最大等待时长。默认值为 `0`，即不进行额外的等待。 |
| [tidb_tmp_table_max_size](/system-variables.md#tidb_tmp_table_max_size-从-v53-版本开始引入) | 新增  | 此变量用于限制单个[临时表](/temporary-tables.md)的最大大小，临时表超出该大小后报错。 |

### 配置文件参数

|  配置文件    |  配置项    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- | :----------- |
| TiDB | [`prepared-plan-cache.capacity`](/tidb-configuration-file.md#capacity)  | 修改 | 此配置项用于控制缓存语句的数量。默认值从 `100` 修改为 `1000`。 |
| TiKV | [`storage.reserve-space`](/tikv-configuration-file.md#reserve-space) | 修改 | 此配置项用于控制 TiKV 启动时用于保护磁盘的预留空间。从 v5.3.0 起，预留空间的 80% 用作磁盘空间不足时运维操作所需要的额外磁盘空间，剩余的 20% 为磁盘临时文件。 |
| TiKV | `memory-usage-limit` | 修改  | 以前的版本没有 memory-usage-limit 参数， 升级后该参数值根据 storage.block-cache.capacity 来计算。 |

> |  TiKV | `raftstore.raft-reject-transfer-leader-duration` | 删除 | 控制迁移 leader 到新加节点的最小时间。|

Reviewer 10: ```suggestion
| TiKV | [`raftstore.store-io-pool-size`](/tikv-configuration-file.md#store-io-pool-size)  | 新增 | 此配置项用于处理 Raft I/O 任务的线程池中线程的数量，`0` 表示 I/O 任务由 Raftstore 线程处理。|
| TiKV | [`raftstore.raft-write-size-limit`](/tikv-configuration-file.md#raft-write-size-limit)  | 新增 | 此配置项用于触发 Raft 数据写入的阈值。当数据大小超过该配置项值，数据会被写入磁盘。当 `raftstore.store-io-pool-size` 的值为 `0` 时，该配置项不生效。|
| TiKV | [`raftstore.raft-msg-flush-interval`](/tikv-configuration-file.md#raft-msg-flush-interval)  | 新增 | Raft 消息攒批发出的间隔时间。每隔该配置项指定的间隔，Raft 消息会攒批发出。当 `raftstore.store-io-pool-size` 的值为 `0` 时，该配置项不会生效。|
|  TiKV | `raftstore.raft-reject-transfer-leader-duration` | 删除 | 控制迁移 leader 到新加节点的最小时间。|
```

> |  TiKV | `raftstore.raft-reject-transfer-leader-duration` | 删除 | 控制迁移 leader 到新加节点的最小时间。|

Reviewer 10: ```suggestion
|  TiKV | `raftstore.raft-reject-transfer-leader-duration` | 删除 | 控制迁移 leader 到新加节点的最小时间。|
|  TiKV | [`raftstore.raft-write-size-limit`](/tikv-configuration-file.md#raft-write-size-limit)  | 新增 | 此配置项用于触发 Raft 数据写入的阈值。当数据大小超过该配置项值，数据会被写入磁盘。当 `raftstore.store-io-pool-size` 的值为 `0` 时，该配置项不生效。|
|  TiKV | [`raftstore.raft-msg-flush-interval`](/tikv-configuration-file.md#raft-msg-flush-interval)  | 新增 | Raft 消息攒批发出的间隔时间。每隔该配置项指定的间隔，Raft 消息会攒批发出。当 `raftstore.store-io-pool-size` 的值为 `0` 时，该配置项不会生效。|
|  TiKV | `raftstore.raft-reject-transfer-leader-duration` | 删除 | 控制迁移 leader 到新加节点的最小时间。|
```

Reviewer 10: 新增项挪到了上面，删除项挪到了最后

Reviewer 11: |  TiKV | `raftstore.store-pool-size` | 含义更改 | v5.3 以前，为 Raft 的线程池中处理消息的线程数量。自 v5.3 起，表示 Raftstore 线程池中执行 CPU 任务的线程数量。|
| TiKV | `raftstore.store-io-pool-size` | 新增 |  表示处理 Raft I/O 任务的线程池中线程的数量，即 StoreWriter 线程池的大小。| 
|  TiKV | `raftstore.raft-write-size-limit` | 新增 | 触发 Raft 数据写入的阈值。当数据大小超过该配置项值，数据会被写入磁盘。当 `raftstore.store-io-pool-size` 的值为 `0` 时，该配置项不生效。|
|  TiKV | `raftstore.raft-msg-flush-interval` | 新增 | Raft 消息攒批发出的间隔时间。每隔该配置项指定的间隔，Raft 消息会攒批发出。当 `raftstore.store-io-pool-size` 的值为 `0` 时，该配置项不生效。|
|  TiKV | `raftstore.raft-reject-transfer-leader-duration` | 删除 | 控制迁移 leader 到新加节点的最小时间。|

    [用户文档](/tidb-lightning/tidb-lightning-prechecks.md)

- **TiDB Lightning 支持导入 GBK 编码的文件**

通过指定源数据文件的字符集，TiDB Lightning 会在导入过程中将源文件从指定的字符集转换为 UTF-8 编码。

    [用户文档](/tidb-lightning/tidb-lightning-configuration.md)

- **Sync-diff-inspector 优化**

    - 大幅提升了对比速度，由原来的 375 MB/s 提升至 700 MB/s
    - 对比过程中对 TiDB 节点的内存消耗降低近一半
    - 优化了用户交互界面，在对比过程中可以显示进度

     [用户文档](/sync-diff-inspector/sync-diff-inspector-overview.md)

### 问题诊断效率

- **保存和恢复集群现场信息**

    在定位排查 TiDB 集群问题时，用户经常需要提供系统和查询计划相关的信息。为了帮助用户更方便地获取相关信息，更高效地排查集群问题，TiDB 在 v5.3.0 中引入了 `PLAN REPLAYER` 命令，用于“一键”保存和恢复现场问题的相关信息，提升查询计划问题诊断的效率，同时方便将问题归档管理。

    `PLAN REPLAYER` 主要功能如下：

    - 导出排查现场 TiDB 集群的相关信息，导出为 ZIP 格式的文件用于保存。
    - 在任意 TiDB 集群上导入另一 TiDB 集群现场信息的 ZIP 文件。

    [用户文档](/sql-plan-replayer.md)，[#26325](https://github.com/pingcap/tidb/issues/26325)

### TiDB 数据共享订阅

- **TiCDC 支持灾备场景下的最终一致性复制**

    在主从灾备架构下，当生产集群（即 TiCDC 同步的上游集群）发生灾难、且短时间内无法恢复对外提供服务时，TiCDC 具备保证从集群数据一致性的能力，并允许业务快速的将流量切换至从集群，避免数据库长时间不可用而对业务造成影响。

    该功能支持 TiCDC 将 TiDB 集群的增量数据复制到备用关系型数据库 TiDB/Aurora/MySQL/MariaDB，在 TiCDC 正常同步没有延迟的情况下，上游发生灾难后，可以在 30 分钟内将下游集群恢复到上游的某个 snapshot 状态，并且允许丢失的数据小于 5 分钟。即 RPO <= 30min，RTO <= 5min。

    [用户文档](/ticdc/manage-ticdc.md)

>     **TiCDC 支持 HTTP 协议 OpenAPI 对 TiCDC 任务进行管理**
> 
>     从 TiDB v5.2.0 起，TiCDC 提供的 OpenAPI 功能成为正式特性，用户可通过 OpenAPI 对 TiCDC 集群进行查询和运维操作。

Author: ```suggestion
- **TiCDC 支持 HTTP 协议 OpenAPI 对 TiCDC 任务进行管理**

    从 TiDB v5.3.0 起，TiCDC 提供的 OpenAPI 功能成为正式特性，用户可通过 OpenAPI 对 TiCDC 集群进行查询和运维操作。
```

    从 TiDB v5.2.0 起，TiCDC 提供的 OpenAPI 功能成为正式特性，用户可通过 OpenAPI 对 TiCDC 集群进行查询和运维操作。

    [用户文档](/ticdc/ticdc-open-api.md)

### 部署及运维

- **持续性能分析（实验特性）**

    TiDB Dashboard 引入持续性能分析功能，提供在集群运行状态时自动保存实例性能分析结果的能力，通过火焰图的形式提高了 TiDB 集群性能的可观测性，有助于缩短故障诊断时间。

    该功能默认关闭，需进入 TiDB Dashboard 持续性能分析页面开启。

    持续性能分析功能必须使用 TiUP 1.7.0 及以上版本升级或安装的集群才可使用。

    [用户文档](/dashboard/continuous-profiling.md)

## 遥测

> TiDB 在遥测中新增收集 <列出本次新增遥测内容>。

Reviewer 3: ```suggestion
TiDB 在遥测中新增收集 temporaryTable 功能的开启情况。
```

> TiDB 在遥测中新增收集 TEMPORARY TABLE 功能的开启情况。

Reviewer 3: ```suggestion
TiDB 在遥测中新增收集 TEMPORARY TABLE 功能的开启情况。收集的数据中不包含任何实际业务的表名或表数据。
```

    - 修复当导出带有 `new collation` 数据的表的统计信息时报 `data too long` 错误的问题 [27024](https://github.com/pingcap/tidb/issues/27024)
    - 修复 `TIDB_TRX` 中不包含重试事务的问题 [28670](https://github.com/pingcap/tidb/pull/28670)
    - 修复配置项 `plugin_dir` 的默认值错误问题 [28084](https://github.com/pingcap/tidb/issues/28084)

+ TiKV

    - 修复 Region 迁移时 Raftstore 模块出现死锁导致 TiKV 不可用的问题。用户可通过关闭调度并重启出问题的 TiKV 来临时应对。[#10909](https://github.com/tikv/tikv/issues/10909)
    - 修复因 Congest 错误而导致的 CDC 频繁增加 scan 重试的问题 [#11082](https://github.com/tikv/tikv/issues/11082)
    - 修复因 channel 打满而导致的 Raft 断连情况 [#11047](https://github.com/tikv/tikv/issues/11047)
    - 修复 Raft client 中 batch 消息过大的问题 [#9714](https://github.com/tikv/tikv/issues/9714)
    - 修复 `resolved_ts` 中协程泄漏的问题 [#10965](https://github.com/tikv/tikv/issues/10965)
    - 修复当 response 大小超过 4 GiB 时 Coprocessor panic 的问题 [#9012](https://github.com/tikv/tikv/issues/9012)
    - 修复当一个 snapshot 文件无法被垃圾清理 (GC) 时 snapshot GC 会缺失 GC snapshot 文件的问题 [#10813](https://github.com/tikv/tikv/issues/10813)
    - 修复当处理 Coprocessor 请求时因超时而导致 panic 的问题 [#10852](https://github.com/tikv/tikv/issues/10852)
    - 修复因统计线程监控数据导致的内存泄漏 [#11195](https://github.com/tikv/tikv/issues/11195)

>     - 修复在某些平台获取 cgroup 信息导致 panic 的问题 [#10980](https://github.com/tikv/tikv/pull/10980)

Reviewer 13: ```suggestion
    - 修复 Compaction Filter GC 无法清除 MVCC Deletion 版本导致 scan 性能下降的问题 [#10980](https://github.com/tikv/tikv/pull/11248)
```

---
title: TiDB 5.3 Release Notes
---

# TiDB 5.3 Release Notes

> 发版日期：2021 年 11 月 30 日

Author: ```suggestion
发版日期：2021 年 11 月 26 日
```

| [`tidb_enable_pseudo_for_outdated_stats`](/system-variables.md#tidb_enable_pseudo_for_outdated_stats-从-v530-版本开始引入) | 新增 | 此变量用于控制优化器在一张表上的统计信息过期时的行为。默认值为 `ON`，当表数据被修改的行数大于该表总行数的 80% （该比例可通过 [`pseudo-estimate-ratio`](/tidb-configuration-file.md#pseudo-estimate-ratio) 配置项调整） 时，优化器认为该表上除总行数以外的统计信息不再可靠，转而使用 pseudo 统计信息。将该变量值设为 `OFF` 后，即使统计信息过期，优化器也仍会使用该表上的统计信息。|
|[`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) | 新增  | 此变量用于开启或关闭 TSO Follower Proxy 特性。默认值为 `OFF`，代表关闭 TSO Follower Proxy 特性。此时，TiDB 仅会从 PD leader 获取 TSO。当开启该特性之后，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而降低 PD leader 的 CPU 压力。 |
|[`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) | 新增 | 此变量用于设置 TiDB 向 PD 请求 TSO 时进行一次攒批操作的最大等待时长。默认值为 `0`，即不进行额外的等待。 |
| [tidb_tmp_table_max_size](/system-variables.md#tidb_tmp_table_max_size-从-v53-版本开始引入) | 新增  | 此变量用于限制单个[临时表](/temporary-tables.md)的最大大小，临时表超出该大小后报错。 |

### 配置文件参数

|  配置文件    |  配置项    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- | :----------- |

> | TiDB | [`prepared-plan-cache.capacity`](/tidb-configuration-file.md#capacity)  | 修改 | 此配置项用于控制缓存语句的数量。默认值从 `100` 修改为 `1000`。 |

Reviewer 10: ```suggestion
| TiDB | [`prepared-plan-cache.capacity`](/tidb-configuration-file.md#capacity)  | 修改 | 此配置项用于控制缓存语句的数量。默认值从 `100` 修改为 `1000`。 |
| TiKV | [`store-io-pool-size`](/tikv-configuration-file.md#store-io-pool-size)  | 新增 | 此配置项用于处理 Raft I/O 任务的线程池中线程的数量，`0` 表示 I/O 任务由 Raftstore 线程处理。|
| TiKV | [`raft-write-size-limit`](/tikv-configuration-file.md#raft-write-size-limit)  | 新增 | 此配置项用于触发 Raft 数据写入的阈值。当数据大小超过该配置项值，数据会被写入磁盘。当 `store-io-pool-size` 的值为 `0` 时，该配置项不生效。|
| TiKV | [`raft-msg-flush-interval`](/tikv-configuration-file.md#raft-msg-flush-interval)  | 新增 | Raft 消息攒批发出的间隔时间。每隔该配置项指定的间隔，Raft 消息会攒批发出。当 `store-io-pool-size` 的值为 `0` 时，该配置项不会生效。|
```

Author: 请把配置项对应的大项也写上，比如 store-pool-size，可以写成 raftstore.store-pool-size

| PD | [`log.file.max-backups`](/pd-configuration-file.md#max-backups) | 修改 | 此配置项用于控制日志文件保留的最大个数。默认值从 `7` 修改为 `0`。 |
| PD | [`patrol-region-interval`](/pd-configuration-file.md#patrol-region-interval) | 修改 | 此配置项用于控制 replicaChecker 检查 Region 健康状态的运行频率，越短则运行越快，通常状况不需要调整。默认值从 `100ms` 修改为 `10ms`。 |
| PD | [`max-snapshot-count`](/pd-configuration-file.md#max-snapshot-count) | 修改 | 此配置项用于控制单个 store 最多同时接收或发送的 snapshot 数量，调度受制于这个配置来防止抢占正常业务的资源。默认值从 `3` 修改为 `64`。 |
| PD | [`max-pending-peer-count`](/pd-configuration-file.md#max-pending-peer-count) | 修改 | 此配置项用于控制单个 store 的 pending peer 上限，调度受制于这个配置来防止在部分节点产生大量日志落后的 Region。默认值从 `16` 修改为 `64`。 |

### 其他

- 临时表：

    - 对于本地临时表，如果在 v5.3.0 升级前创建了本地临时表，这些临时表实际为普通表，在升级至 v5.3.0 或更高版本后，也会被 TiDB 当成普通表处理。对于全局临时表，如果在 v5.3.0 上创建了全局临时表，当 TiDB 降级至 v5.3.0 以前版本后，这些临时表会被当作普通表处理，导致数据错误。
    - TiCDC 和 BR 从 v5.3.0 开始支持[全局临时表](/temporary-tables.md#全局临时表)。如果使用 v5.3.0 以下版本同步全局临时表到下游，会导致表定义错误。
    - 通过 TiDB 生态工具导入的集群、恢复后的集群、同步的下游集群必须是 TiDB v5.3.0 及以上版本，否则创建全局临时表时报错。
    - 关于临时表的更多兼容性信息，请参考 [与 MySQL 临时表的兼容性](/temporary-tables.md#与-mysql-临时表的兼容性)和[与其他 TiDB 功能的兼容性限制](/temporary-tables.md#与其他-tidb-功能的兼容性限制)。

> 

Author: ```suggestion

- 对于 v5.3.0 之前的版本，当系统变量设置为非法值时，TiDB 会报错。从 v5.3.0 起，当系统变量设置为非法值时，TiDB 会返回成功，并报类似 “|Warning | 1292 | Truncated incorrect xxx: 'xx' ” 的警告。
```

Author: @djshow832 PTAL Thanks

    建议为访问 TiDB Dashboard 创建一个最小权限的 SQL 用户，并用该用户登录 TiDB Dashboard，避免使用高权限用户，提升安全性。

    [用户文档](/dashboard/dashboard-user.md)

### 性能

- **优化 PD 时间戳处理流程**

    优化 TiDB 时间戳处理流程，支持通过开启 PD Follower Proxy 和调整 PD client 批量请求 TSO 时所需的 batch 等待时间的方式来降低 PD leader 时间戳处理负载，提升系统整体可扩展性。

    - 支持通过系统变量 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) 设置 PD Follower Proxy 功能开关。在 PD 时间戳请求负载过高的情况下，通过开启 PD Follower Proxy，可以将 follower 上请求周期内收集到的 TSO request 批量转发到 leader 节点，从而有效减少 client 与 leader 的直接交互次数，降低 leader 的负载，提升 TiDB 整体性能。

    > **注意：**
    >
    > 在 client 数较少、PD leader 负载不高的情况下，不建议开启 PD Follower Proxy 功能。

    - 支持通过 [`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) 系统变量设置 PD client 批量请求 TSO 时所需的最大 batch 等待时间，单位为毫秒。在 PD TSO 请求负载过高的情况下，可以通过调大等待时间获得更大的 batch size，从而降低 PD 负载，提升吞吐。

    > **注意：**
    >
    > 在 TSO 请求负载不高的情况下，不建议调整该参数。

    [用户文档](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入)，[#3149](https://github.com/tikv/pd/issues/3149)

### 稳定性

- **支持多节点永久损坏后的在线有损恢复（实验特性）**

>     新增对 `unsafe remove-failed-stores` 命令的支持，实现数据有损恢复。当多数副本发生了永久性损坏（如磁盘损坏）等问题，导致无法在业务层读写一段数据时，PD 可以执行在线数据恢复，使该数据恢复至可读写状态。

Reviewer 13: ```suggestion
    新增对 `pd-ctl unsafe remove-failed-stores` 命令的支持，实现在线数据有损恢复。当多数副本发生了永久性损坏（如磁盘损坏）等问题，导致无法在业务层读写一段数据时，PD 可以执行在线数据恢复，使该数据恢复至可读写状态。
```

    进行该功能相关操作时，建议在 TiDB 团队支持下完成。

    [用户文档](/online-unsafe-recovery.md)，[#10483](https://github.com/tikv/tikv/issues/10483)

### 数据迁移

- **提高 DM 同步性能**

    支持以下功能，实现以更低的延迟将数据从 MySQL 同步数据到 TiDB。

    - 合并单行数据的多次变更（Compact multiple updates on a single row into one statement）
    - 点查更新合并为批量操作（Merge batch updates of multiple rows into one statement）

- **增加 DM 的 OpenAPI 以更方便地管理集群（实验特性）**

    DM 提供 OpenAPI 功能，用户可通过 OpenAPI 对 DM 集群进行查询和运维操作。OpenAPI 的总体功能和 [dmctl 工具](https://docs.pingcap.com/zh/tidb-data-migration/stable/dmctl-introduction)类似。

    当前 OpenAPI 功能为实验特性，默认关闭，不建议在生产环境中使用。

    [用户文档](https://docs.pingcap.com/zh/tidb-data-migration/stable/open-api)

- **TiDB Lightning 并行导入**

    TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。该功能无需特别的配置，在不改变用户使用习惯的同时，极大提高了用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。

经测试，使用 10 台 TiDB Lightning，20 TB 规模的 MySQL 数据可以在 8 小时内导入到 TiDB，单台 TiDB Lightning 可以支持 250 GB/s 的导入速度，整体效率提升了 8 倍。

> 

Reviewer 14: ```suggestion
```

- v5.3.0之前，对于新增索引，analyze 时间不受设定时间的限制，tidb_auto_analyze_start_time 和 tidb_auto_analyze_end_time 时间段内将不会出发 auto analyze
- plugin 的默认路径从 "" 改为 /data/deploy/plugin

## 新功能

### SQL

- **SQL 接口设置数据放置规则（实验特性）**

    增加 `[CREATE | ALTER] PLACEMENT POLICY` 语句支持，提供 SQL 接口设置数据放置规则。通过该功能，用户可以指定任意连续数据按照不同地域，机房，机柜，主机，硬件，副本数规则进行部署，满足低成本，高可用，灵活多变的业务诉求。该功能可以实现以下业务场景：

    - 跨区域放置数据以改善局部访问性能
    - 多业务数据库合并，降低大量数据库的常规运维管理成本，并通过规则配置实现业务资源隔离
    - 增加更重要数据的副本数，提高业务可用性，数据可靠性
    - 将最新数据存入 SSD，历史数据存入 HDD，降低归档数据存储成本
    - 把热点数据的 leader 放到高性能的 TiKV 实例上
    - 将不相关的数据分离到不同的存储中以提高可用性

    [用户文档](/)

- **临时表**

    增加 CREATE [GLOBAL] TEMPORARY TABLE 语句支持，支持创建临时表，方便业务管理中间计算的临时数据。临时表中的数据均保存在内存中，用户可通过 tidb_tmp_table_max_size 变量限制临时表的内存大小。TiDB 支持以下两种临时表：

    - Global 临时表
        - 对集群内所有 session 可见，表结构持久化。
        - 提供事务级别的数据隔离，数据只在事务内有效，事务结束后自动删除数据。
    - Local 临时表
        - 只对当前 session 可见，表结构不持久化。
        - 支持重名，用户无需为业务设计复杂的表命名规则。
        - 提供会话级别的数据隔离，降低业务设计复杂度，会话结束后删除临时表。

    [用户文档](/)

- **支持 `FOR UPDATE OF TABLES` 语法**

    对于存在多表 join 的语句，支持只对 `OF TABLES` 中包含的表关联的行进行悲观锁加锁操作。

    [用户文档](/sql-statement-select.md), [#28689](https://github.com/pingcap/tidb/issues/28689)

> - **表/分区表属性设置**

Reviewer 16: ```suggestion
- **表属性设置**
```

> - **支持 `FOR UPDATE OF TABLES` 语法**

Reviewer 11: /sql-statement-select.md 文档里没有写 `FOR UPDATE OF TABLES` 的介绍。
能请 @coocood @cfzjywxk 添加一下吗？

Author: https://github.com/pingcap/docs-cn/pull/7412/files 这个算不算

Reviewer 11: Get. 谢谢~

    - 修复 `resolved_ts` 中协程泄漏的问题 [#10965](https://github.com/tikv/tikv/issues/10965)
    - 修复当 response 大小超过 4 GiB 时 Coprocessor panic 的问题 [#9012](https://github.com/tikv/tikv/issues/9012)
    - 修复当一个 snapshot 文件无法被垃圾清理 (GC) 时 snapshot GC 会缺失 GC snapshot 文件的问题 [#10813](https://github.com/tikv/tikv/issues/10813)
    - 修复当处理 Coprocessor 请求时因超时而导致 panic 的问题 [#10852](https://github.com/tikv/tikv/issues/10852)
    - 修复因统计线程监控数据导致的内存泄漏 [#11195](https://github.com/tikv/tikv/issues/11195)
    - 修复在某些平台获取 cgroup 信息导致 panic 的问题 [#10980](https://github.com/tikv/tikv/pull/10980)
+ PD

    - 修复因超过副本配置数量而导致错误删除带有数据且处于 pending 状态的副本的问题 [#4045](https://github.com/tikv/pd/issues/4045)
    - 修复 PD 未能及时修复 Down Peer 副本的问题 [#4077](https://github.com/tikv/pd/issues/4077)
    - 修复 Scatter Range 调度器无法对空 Region 进行调度的问题 [#4118](https://github.com/tikv/pd/pull/4118)
    - 修复 key manager 占用过多 CPU 的问题 [#4071](https://github.com/tikv/pd/issues/4071)
    - 修复热点调度器变更配置的过程中可能会存在的数据竞争问题 [#4159](https://github.com/tikv/pd/issues/4159)
    - 修复因 Region syncer 卡住而导致 leader 选举慢的问题 [#3936](https://github.com/tikv/pd/issues/3936)

+ TiFlash

> 

Reviewer 17: ```suggestion

    - 修复 TiFlash Store Size 统计结果不准确的问题
```

    - 对比过程中对 TiDB 节点的内存消耗降低近一半
    - 优化了用户交互界面，在对比过程中可以显示进度
   
     [用户文档](/sync-diff-inspector/sync-diff-inspector-overview.md)

### 问题诊断效率

- **保存和恢复集群现场信息**

    在定位排查 TiDB 集群问题时，用户经常需要提供系统和查询计划相关的信息。为了帮助用户更方便地获取相关信息，更高效地排查集群问题，TiDB 在 v5.3.0 中引入了 `PLAN REPLAYER` 命令，用于“一键”保存和恢复现场问题的相关信息，提升查询计划问题诊断的效率，同时方便将问题归档管理。

    `PLAN REPLAYER` 主要功能如下：

    - 导出排查现场 TiDB 集群的相关信息，导出为 ZIP 格式的文件用于保存。
    - 在任意 TiDB 集群上导入另一 TiDB 集群现场信息的 ZIP 文件。

    [用户文档](/sql-plan-replayer.md), [#26325](https://github.com/pingcap/tidb/issues/26325)

### TiDB 数据共享订阅

- **TiCDC 支持灾备场景下的最终一致性复制**

    在主从灾备架构下，当生产集群（即 TiCDC 同步的上游集群）发生灾难、且短时间内无法恢复对外提供服务时，TiCDC 具备保证从集群数据一致性的能力，并允许业务快速的将流量切换至从集群，避免数据库长时间不可用而对业务造成影响。

    该功能支持 TiCDC 将 TiDB 集群的增量数据复制到备用关系型数据库 TiDB/Aurora/MySQL/MariaDB，在 TiCDC 正常同步没有延迟的情况下，上游发生灾难后，可以在 30 分钟内将下游集群恢复到上游的某个 snapshot 状态，并且允许丢失的数据小于 5 分钟。即 RPO <= 30min，RTO <= 5min。

    [用户文档](/ticdc/manage-ticdc.md)  

### 部署及运维

- **持续性能分析（实验特性）**

    TiDB Dashboard 引入持续性能分析功能，提供在集群运行状态时自动保存实例性能分析结果的能力，通过火焰图的形式提高了 TiDB 集群性能的可观测性，有助于缩短故障诊断时间。

    该功能默认关闭，需进入 TiDB Dashboard 持续性能分析页面开启。

    持续性能分析功能必须使用 TiUP 1.7.0 及以上版本升级或安装的集群才可使用。

    [用户文档](/dashboard/continuous-profiling.md)  

### 遥测

> TiDB 在遥测中新增收集 <列出本次新增遥测内容>。

Author: @lilyjazz  可以帮忙确认下 v5.3 遥测范围有变化吗？谢谢

Reviewer 3: 遥测没有变化

Reviewer 3: 在11月23日，确定遥测数据有新增。

|[`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) | 新增  | 此变量用于开启或关闭 TSO Follower Proxy 特性。默认值为 `OFF`，代表关闭 TSO Follower Proxy 特性。此时，TiDB 仅会从 PD leader 获取 TSO。当开启该特性之后，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而降低 PD leader 的 CPU 压力。 |
|[`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) | 新增 | 此变量用于设置 TiDB 向 PD 请求 TSO 时进行一次攒批操作的最大等待时长。默认值为 `0`，即不进行额外的等待。 |
| [tidb_tmp_table_max_size](/system-variables.md#tidb_tmp_table_max_size-从-v53-版本开始引入) | 新增  | 此变量用于限制单个[临时表](/temporary-tables.md)的最大大小，临时表超出该大小后报错。 |

### 配置文件参数

|  配置文件    |  配置项    |  修改类型    |  描述    |
| :---------- | :----------- | :----------- | :----------- |
| TiDB | [`prepared-plan-cache.capacity`](/tidb-configuration-file.md#capacity)  | 修改 | 此配置项用于控制缓存语句的数量。默认值从 `100` 修改为 `1000`。 |
| TiKV | [`storage.reserve-space`](/tikv-configuration-file.md#reserve-space) | 修改 | 此配置项用于控制 TiKV 启动时用于保护磁盘的预留空间。从 v5.3.0 起，预留空间的 80% 用作磁盘空间不足时运维操作所需要的额外磁盘空间，剩余的 20% 为磁盘临时文件。 |
| TiKV | `memory-usage-limit` | 修改  | 以前的版本没有 memory-usage-limit 参数， 升级后该参数值根据 storage.block-cache.capacity 来计算。 |

> |  TiKV | `raftstore.store-pool-size` | 含义更改 | v5.3 以前，为 Raft 的线程池中处理消息的线程数量。自 v5.3 起，表示 Raftstore 线程池中执行 CPU 任务的线程数量。|

Reviewer 10: ```suggestion
```

    - 支持通过 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) 系统变量设置 PD follower proxy 功能开关。在 PD 时间戳请求负载过高的情况下，通过开启 PD follower proxy，可以将 follower 上请求周期内收集到的 TSO request 批量转发到 leader 节点，有效减少 client 与 leader 的直接交互次数，降低 leader 的负载，提升 TiDB 整体性能。

    > **注意：**
    >
    > 在 client 数较少的情况下，PD leader 负载不高的情况下，不建议开启 PD follower proxy 功能。

    - 支持通过 [`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) 系统变量设置 PD client 批量请求 TSO 的最大等待时间，单位毫秒。在 PD TSO 请求负载过高的情况下，通过调大该参数，可以提升一次请求 TSO 的数量，降低 PD 负载，提升吞吐。

    > **注意：**
    >
    > 在 TSO 请求负载不高的情况下，不建议调整该参数。

    [用户文档](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入), [#3149](https://github.com/tikv/pd/issues/3149)

### 稳定性

- **支持 Raft 多数副本丢失时数据的在线恢复能力（实验特性）**

    增加 PD-CTL RECOVER-STORE-FAILURE 语句支持，提供在线恢复 TiKV 实例能力。通过该功能，可以实现：

    - 在线恢复所有 Raft 多数副本丢失的 Region 可读写。
    - 保证恢复所有 Region 后没有数据空洞。

    需要注意的是，Raft 多数副本失败的情况下无法避免已提交数据的丢失。

    [用户文档](/online-unsafe-recovery.md), [#10483](https://github.com/tikv/tikv/issues/10483)

### 数据迁移

> - **提高 DM 复制性能**

Reviewer 15: ```suggestion
- **提高 DM 同步性能**
```

| TiKV | memory-usage-limit | 修改  | 以前的版本没有 memory-usage-limit 参数， 升级后改参数值根据 storage.block-cache.capacity来计算。 |
| PD | [`log.file.max-days`](/pd-configuration-file.md#max-days) | 修改 | 此配置项用于控制日志保留的最长天数。默认值从 `1` 修改为 `0`。 |
| PD | [`log.file.max-backups`](/pd-configuration-file.md#max-backups) | 修改 | 此配置项用于控制日志文件保留的最大个数。默认值从 `7` 修改为 `0`。 |
| PD | [`patrol-region-interval`](/pd-configuration-file.md#patrol-region-interval) | 修改 |此配置项用于控制 replicaChecker 检查 Region 健康状态的运行频率，越短则运行越快，通常状况不需要调整。默认值从 `100ms` 修改为 `10ms`。 |
| PD | [`max-snapshot-count`](/pd-configuration-file.md#max-snapshot-count) | 修改 |此配置项用于控制单个 store 最多同时接收或发送的 snapshot 数量，调度受制于这个配置来防止抢占正常业务的资源。默认值从 `3` 修改为 `64`。 |
| PD | [`max-pending-peer-count`](/pd-configuration-file.md#max-pending-peer-count) | 修改 |此配置项用于控制单个 store 的 pending peer 上限，调度受制于这个配置来防止在部分节点产生大量日志落后的 Region。默认值从 `16` 修改为 `64`。 |

### 其他

- 临时表：

    - 如果在 v5.3.0 升级前创建了本地临时表，这些临时表实际为普通表，在升级后也会被 TiDB 当成普通表处理。在 v5.3.0 上创建的全局临时表在降级后会被当作普通表处理，导致数据错误。
    - TiCDC 和 BR 从 v5.3.0 开始支持[全局临时表](/temporary-table.md#全局临时表)。如果使用 v5.3.0 以下版本同步全局临时表到下游，会导致表定义错误。

>     - 通过 TiDB 生态工具导入的集群、恢复后的集群、同步的下游集群必须是 TiDB v5.3.0 及以上版本，否则报错。

Reviewer 6: ```suggestion
    - 通过 TiDB 生态工具导入的集群、恢复后的集群、同步的下游集群必须是 TiDB v5.3.0 及以上版本，否则创建全局临时表时报错。
```

    - 关于临时表的更多兼容性信息，请参考 [与 MySQL 临时表的兼容性](/temporary-table.md#与-mysql-临时表的兼容性)和[与其他 TiDB 功能的兼容性限制](/temporary-table.md#与其他-tidb-功能的兼容性限制)。

- 修正 `SHOW CREATE VIEW` 不需要 `SHOW VIEW` 权限的问题，现在用户必须具有 `SHOW VIEW` 权限才允许执行 `SHOW CREATE VIEW` 语句。

> - 系统变量 `sql_auto_is_null` 被加入 Noop Funciton 中，当 `tidb_enable_noop_functions = 0/OFF` 时，修改改变量会报错。

Reviewer 6: ```suggestion
- 系统变量 `sql_auto_is_null` 被加入 Noop Function 中，当 `tidb_enable_noop_functions = 0/OFF` 时，修改该变量会报错。
```

    > **注意：**
    >
    > 在 client 数较少的情况下，PD leader 负载不高的情况下，不建议开启 PD follower proxy 功能。

    - 支持通过 [`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) 系统变量设置 PD client 批量请求 TSO 的最大等待时间，单位毫秒。在 PD TSO 请求负载过高的情况下，通过调大该参数，可以提升一次请求 TSO 的数量，降低 PD 负载，提升吞吐。

    > **注意：**
    >
    > 在 TSO 请求负载不高的情况下，不建议调整该参数。

    [用户文档](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入), [#3149](https://github.com/tikv/pd/issues/3149)

### 稳定性

- **支持 Raft 多数副本丢失时数据的在线恢复能力（实验特性）**

    增加 PD-CTL RECOVER-STORE-FAILURE 语句支持，提供在线恢复 TiKV 实例能力。通过该功能，可以实现：

    - 在线恢复所有 Raft 多数副本丢失的 Region 可读写。
    - 保证恢复所有 Region 后没有数据空洞。

    需要注意的是，Raft 多数副本失败的情况下无法避免已提交数据的丢失。

    [用户文档](/online-unsafe-recovery.md), [#10483](https://github.com/tikv/tikv/issues/10483)

### 数据迁移

- **提高 DM 复制性能**

    支持以下功能，实现以更低的延迟将数据从 MySQL 同步数据到 TiDB。

    - 合并单行数据的多次变更（Compact multiple updates on a single row into one statement）
    - 点查更新合并为批量操作（Merge batch updates of multiple rows into one statement）

>     - 异步保存检查点（Async Checkpoint）

Reviewer 15: @sunzhaoyang 这个我记得写技术博客的时候去掉了。这里要删去吗？

Reviewer 2: Yes

>     - 异步保存检查点（Async Checkpoint）

Reviewer 2: ```suggestion
```

| [tidb_enable_noop_functions](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) | 修改 | 此变量的控制范围不再包括 `CREATE TEMPORARY TABLE` 和 `DROP TEMPORARY TABLE` 行为。 |
| [`tidb_enable_pseudo_for_outdated_stats`](/system-variables.md#tidb_enable_pseudo_for_outdated_stats-从-v530-版本开始引入) | 新增 | 此变量用于控制优化器在一张表上的统计信息过期时的行为。默认值为 `ON`，当表数据被修改的行数大于该表总行数的 80% （该比例可通过 [`pseudo-estimate-ratio`](/tidb-configuration-file.md#pseudo-estimate-ratio) 配置项调整） 时，优化器认为该表上除总行数以外的统计信息不再可靠，转而使用 pseudo 统计信息。将该变量值设为 `OFF` 后，即使统计信息过期，优化器也仍会使用该表上的统计信息。|
|[`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v53-版本开始引入) | 新增  | 此变量用于开启或关闭 TSO Follower Proxy 特性。默认值为 `OFF`，代表关闭TSO Follower Proxy 特性。此时，TiDB 仅会从 PD leader 获取 TSO。当开启该特性之后，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而降低 PD leader 的 CPU 压力。 |
|[`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入) | 新增 | 此变量用于设置 TiDB 向 PD 请求 TSO 时进行一次攒批操作的最大等待时长。默认值为 `0`，即不进行额外的等待。 |
| [tidb_tmp_table_max_size](/system-variables.md#tidb_tmp_table_max_size-从-v53-版本开始引入) | 新增  | 此变量用于限制单个[临时表](/temporary-table.md)的最大大小，临时表超出该大小后报错。 |

> | `tmp_table_size` | 修改 | 更名为 `tidb_tmp_table_size`，不再保留 `tmp_table_size`。升级前 global 级别tmp_table_size 有值，升级后需要手动转换为 tidb_tmp_table_size。 |

Reviewer 6: ```suggestion
| `tmp_table_size` | 修改 | 继续保留，但不再生效，改用 `tidb_tmp_table_max_size` 限制临时表的大小。 |
```
之前临时表没有 release 过，按理说没人用 `tmp_table_size` 限制过临时表，所以不用声明升级时的操作。

Author: 好的，如果这条不建议用户配置，而且系统变量文档里也不提，那这行可以从 release notes 里删除

    > **注意：**
    >
    > 在 TSO 请求负载不高的情况下，不建议调整该参数。

    [用户文档](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v53-版本开始引入), [#3149](https://github.com/tikv/pd/issues/3149)

### 稳定性

- **支持 Raft 多数副本丢失时数据的在线恢复能力（实验特性）**

    增加 PD-CTL RECOVER-STORE-FAILURE 语句支持，提供在线恢复 TiKV 实例能力。通过该功能，可以实现：

    - 在线恢复所有 Raft 多数副本丢失的 Region 可读写。
    - 保证恢复所有 Region 后没有数据空洞。

    需要注意的是，Raft 多数副本失败的情况下无法避免已提交数据的丢失。

    [用户文档](/online-unsafe-recovery.md), [#10483](https://github.com/tikv/tikv/issues/10483)

### 数据迁移

- **提高 DM 复制性能**

    支持以下功能，实现以更低的延迟将数据从 MySQL 同步数据到 TiDB。

    - 合并单行数据的多次变更（Compact multiple updates on a single row into one statement）
    - 点查更新合并为批量操作（Merge batch updates of multiple rows into one statement）
    - 异步保存检查点（Async Checkpoint）

- **增加 DM 的 OpenAPI 以更方便地管理集群（实验特性）**

    DM 提供 OpenAPI 功能，用户可通过 OpenAPI 对 DM 集群进行查询和运维操作。OpenAPI 的总体功能和 [dmctl 工具](https://docs.pingcap.com/zh/tidb-data-migration/stable/dmctl-introduction)类似。

    当前 OpenAPI 功能为实验特性，默认关闭，不建议在生产环境中使用。

    [用户文档](https://docs.pingcap.com/zh/tidb-data-migration/stable/open-api)

- **TiDB Lightning 分布式并行导入**

>     TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。 该功能无需特别的配置，在不改变用户使用习惯的同时，极大提高了用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。

Reviewer 14: ```suggestion
    TiDB Lightning 支持用户同时部署多个 Lightning，并行地将单表或者多表数据迁移 TiDB。该功能无需特别的配置，在不改变用户使用习惯的同时，极大提高了用户的数据迁移能力，助力大数据量业务架构升级，在生产环境使用 TiDB。
```

TiDB 在遥测中新增收集 <列出本次新增遥测内容>。

若要了解所收集的信息详情及如何禁用该行为，请参见[遥测](/telemetry.md)文档。

## 提升改进

+ TiDB

    - 当 coprocessor 遇到锁时，在调试日志中显示受影响的 SQL 语句帮助诊断问题 [#27718](https://github.com/pingcap/tidb/issues/27718)
    - 在 SQL 逻辑层备份和恢复数据时，支持显示备份和恢复数据的大小 [#27247](https://github.com/pingcap/tidb/issues/27247)

+ TiKV

    - 简化 L0 层流控算法 [#10879](https://github.com/tikv/tikv/issues/10879)
    - 优化 raft client 错误日志的收集 [#10983](https://github.com/tikv/tikv/pull/10983)
    - 优化日志线程以避免其成为性能瓶颈 [#10841](https://github.com/tikv/tikv/issues/10841)
    - 添加更多的写入查询统计类型 [#10507](https://github.com/tikv/tikv/issues/10507)

+ PD

    - 热点调度器的 QPS 维度支持更多的写请求类型 [#3869](https://github.com/tikv/pd/issues/3869)
    - 通过动态调整 Balance Region 调度器的重试上限，优化该调度器的性能 [#3744](https://github.com/tikv/pd/issues/3744)
    - 将 TiDB Dashboard 升级至 v2021.10.08.1 [#4070](https://github.com/tikv/pd/pull/4070)
    - 允许 Evict Leader 调度器调度拥有不健康副本的 Region [#4093](https://github.com/tikv/pd/issues/4093)
    - 优化调度器退出的速度 [#4146](https://github.com/tikv/pd/issues/4146)

> 

Reviewer: + TiFlash

    - 修复 TiFlash 在部分平台上由于缺失 `nsl` 库而无法启动的问题
    - 阻止 `wait index` 无限等待，防止写入压力较重时 TiFlash 长时间等待数据同步而无法提供服务的问题。默认超时为5分钟。
    - 解决了当日志体量很大时，用户搜索日志很慢或搜索不出的问题
    - 解决了搜索比较久远的历史日志时，只能搜索出最近的一部分日志的问题
    - 修复在打开 `new collation` 的情况下可能出现的结果错误
    - 修复 SQL 语句中含有极长嵌套表达式时可能出现的解析错误
    - 修复 Exchange 算子中可能出现的 `Block schema mismatch` 错误
    - 修复 Decimal 类型比较时可能出现的 `Can't compare` 错误
    - 修复 `left/substring` 函数中的 `3rd arguments of function substringUTF8 must be constants` 错误

Author: OK.  I will add the content according to https://github.com/moyun/docs-cn/blob/add_5.3_release_note/releases/release-5.3.0.md

    - 修复因 channel 打满而导致的 Raft 断连情况 [#11047](https://github.com/tikv/tikv/issues/11047)
    - 修复 Raft client 中 batch 消息过大的问题 [#9714](https://github.com/tikv/tikv/issues/9714)
    - 修复 `resolved_ts` 中协程泄漏的问题 [#10965](https://github.com/tikv/tikv/issues/10965)
    - 修复当 response 大小超过 4 GiB 时 Coprocessor panic 的问题 [#9012](https://github.com/tikv/tikv/issues/9012)
    - 修复当一个 snapshot 文件无法被垃圾清理 (GC) 时 snapshot GC 会缺失 GC snapshot 文件的问题 [#10813](https://github.com/tikv/tikv/issues/10813)
    - 修复当处理 Coprocessor 请求时因超时而导致 panic 的问题 [#10852](https://github.com/tikv/tikv/issues/10852)

+ PD

    - 修复因超过副本配置数量而导致错误删除带有数据且处于 pending 状态的副本的问题 [#4045](https://github.com/tikv/pd/issues/4045)
    - 修复 PD 未能及时修复 Down Peer 副本的问题 [#4077](https://github.com/tikv/pd/issues/4077)
    - 修复 Scatter Range 调度器无法对空 Region 进行调度的问题 [#4118](https://github.com/tikv/pd/pull/4118)
    - 修复 key manager 占用过多 CPU 的问题 [#4071](https://github.com/tikv/pd/issues/4071)
    - 修复热点调度器变更配置的过程中可能会存在的数据竞争问题 [#4159](https://github.com/tikv/pd/issues/4159)
    - 修复因 Region syncer 卡住而导致 leader 选举慢的问题 [#3936](https://github.com/tikv/pd/issues/3936)

+ TiFlash

    - 修复 TiFlash 在部分平台上由于缺失 `nsl` 库而无法启动的问题

> 

Reviewer: - 阻止 `wait index` 无限等待，防止写入压力较重时 TiFlash 长时间等待数据同步而无法提供服务的问题。默认超时为5分钟。
    - 解决了当日志体量很大时，用户搜索日志很慢或搜索不出的问题
    - 解决了搜索比较久远的历史日志时，只能搜索出最近的一部分日志的问题
    - 修复在打开 `new collation` 的情况下可能出现的结果错误
    - 修复 SQL 语句中含有极长嵌套表达式时可能出现的解析错误
    - 修复 Exchange 算子中可能出现的 `Block schema mismatch` 错误
    - 修复 Decimal 类型比较时可能出现的 `Can't compare` 错误
    - 修复 `left/substring` 函数中的 `3rd arguments of function substringUTF8 must be constants` 错误

Author: OK.  I will add the content according to https://github.com/moyun/docs-cn/blob/add_5.3_release_note/releases/release-5.3.0.md
